// @pulumi/azure/containerservice/kubernetesCluster.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.KubernetesCluster = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Managed Kubernetes Cluster (also known as AKS / Azure Kubernetes Service)
 *
 * ## Example Usage
 *
 * This example provisions a basic Managed Kubernetes Cluster.
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const exampleResourceGroup = new azure.core.ResourceGroup("exampleResourceGroup", {location: "West Europe"});
 * const exampleKubernetesCluster = new azure.containerservice.KubernetesCluster("exampleKubernetesCluster", {
 *     location: exampleResourceGroup.location,
 *     resourceGroupName: exampleResourceGroup.name,
 *     dnsPrefix: "exampleaks1",
 *     defaultNodePool: {
 *         name: "default",
 *         nodeCount: 1,
 *         vmSize: "Standard_D2_v2",
 *     },
 *     identity: {
 *         type: "SystemAssigned",
 *     },
 *     tags: {
 *         Environment: "Production",
 *     },
 * });
 * export const clientCertificate = exampleKubernetesCluster.kubeConfigs.apply(kubeConfigs => kubeConfigs[0].clientCertificate);
 * export const kubeConfig = exampleKubernetesCluster.kubeConfigRaw;
 * ```
 *
 * ## Import
 *
 * Managed Kubernetes Clusters can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:containerservice/kubernetesCluster:KubernetesCluster cluster1 /subscriptions/00000000-0000-0000-0000-000000000000/resourcegroups/group1/providers/Microsoft.ContainerService/managedClusters/cluster1
 * ```
 */
class KubernetesCluster extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            inputs["addonProfile"] = state ? state.addonProfile : undefined;
            inputs["apiServerAuthorizedIpRanges"] = state ? state.apiServerAuthorizedIpRanges : undefined;
            inputs["autoScalerProfile"] = state ? state.autoScalerProfile : undefined;
            inputs["automaticChannelUpgrade"] = state ? state.automaticChannelUpgrade : undefined;
            inputs["defaultNodePool"] = state ? state.defaultNodePool : undefined;
            inputs["diskEncryptionSetId"] = state ? state.diskEncryptionSetId : undefined;
            inputs["dnsPrefix"] = state ? state.dnsPrefix : undefined;
            inputs["dnsPrefixPrivateCluster"] = state ? state.dnsPrefixPrivateCluster : undefined;
            inputs["enablePodSecurityPolicy"] = state ? state.enablePodSecurityPolicy : undefined;
            inputs["fqdn"] = state ? state.fqdn : undefined;
            inputs["httpProxyConfig"] = state ? state.httpProxyConfig : undefined;
            inputs["identity"] = state ? state.identity : undefined;
            inputs["kubeAdminConfigRaw"] = state ? state.kubeAdminConfigRaw : undefined;
            inputs["kubeAdminConfigs"] = state ? state.kubeAdminConfigs : undefined;
            inputs["kubeConfigRaw"] = state ? state.kubeConfigRaw : undefined;
            inputs["kubeConfigs"] = state ? state.kubeConfigs : undefined;
            inputs["kubeletIdentities"] = state ? state.kubeletIdentities : undefined;
            inputs["kubernetesVersion"] = state ? state.kubernetesVersion : undefined;
            inputs["linuxProfile"] = state ? state.linuxProfile : undefined;
            inputs["localAccountDisabled"] = state ? state.localAccountDisabled : undefined;
            inputs["location"] = state ? state.location : undefined;
            inputs["maintenanceWindow"] = state ? state.maintenanceWindow : undefined;
            inputs["name"] = state ? state.name : undefined;
            inputs["networkProfile"] = state ? state.networkProfile : undefined;
            inputs["nodeResourceGroup"] = state ? state.nodeResourceGroup : undefined;
            inputs["portalFqdn"] = state ? state.portalFqdn : undefined;
            inputs["privateClusterEnabled"] = state ? state.privateClusterEnabled : undefined;
            inputs["privateClusterPublicFqdnEnabled"] = state ? state.privateClusterPublicFqdnEnabled : undefined;
            inputs["privateDnsZoneId"] = state ? state.privateDnsZoneId : undefined;
            inputs["privateFqdn"] = state ? state.privateFqdn : undefined;
            inputs["privateLinkEnabled"] = state ? state.privateLinkEnabled : undefined;
            inputs["publicNetworkAccessEnabled"] = state ? state.publicNetworkAccessEnabled : undefined;
            inputs["resourceGroupName"] = state ? state.resourceGroupName : undefined;
            inputs["roleBasedAccessControl"] = state ? state.roleBasedAccessControl : undefined;
            inputs["servicePrincipal"] = state ? state.servicePrincipal : undefined;
            inputs["skuTier"] = state ? state.skuTier : undefined;
            inputs["tags"] = state ? state.tags : undefined;
            inputs["windowsProfile"] = state ? state.windowsProfile : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.defaultNodePool === undefined) && !opts.urn) {
                throw new Error("Missing required property 'defaultNodePool'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["addonProfile"] = args ? args.addonProfile : undefined;
            inputs["apiServerAuthorizedIpRanges"] = args ? args.apiServerAuthorizedIpRanges : undefined;
            inputs["autoScalerProfile"] = args ? args.autoScalerProfile : undefined;
            inputs["automaticChannelUpgrade"] = args ? args.automaticChannelUpgrade : undefined;
            inputs["defaultNodePool"] = args ? args.defaultNodePool : undefined;
            inputs["diskEncryptionSetId"] = args ? args.diskEncryptionSetId : undefined;
            inputs["dnsPrefix"] = args ? args.dnsPrefix : undefined;
            inputs["dnsPrefixPrivateCluster"] = args ? args.dnsPrefixPrivateCluster : undefined;
            inputs["enablePodSecurityPolicy"] = args ? args.enablePodSecurityPolicy : undefined;
            inputs["httpProxyConfig"] = args ? args.httpProxyConfig : undefined;
            inputs["identity"] = args ? args.identity : undefined;
            inputs["kubeletIdentities"] = args ? args.kubeletIdentities : undefined;
            inputs["kubernetesVersion"] = args ? args.kubernetesVersion : undefined;
            inputs["linuxProfile"] = args ? args.linuxProfile : undefined;
            inputs["localAccountDisabled"] = args ? args.localAccountDisabled : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["maintenanceWindow"] = args ? args.maintenanceWindow : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["networkProfile"] = args ? args.networkProfile : undefined;
            inputs["nodeResourceGroup"] = args ? args.nodeResourceGroup : undefined;
            inputs["privateClusterEnabled"] = args ? args.privateClusterEnabled : undefined;
            inputs["privateClusterPublicFqdnEnabled"] = args ? args.privateClusterPublicFqdnEnabled : undefined;
            inputs["privateDnsZoneId"] = args ? args.privateDnsZoneId : undefined;
            inputs["privateLinkEnabled"] = args ? args.privateLinkEnabled : undefined;
            inputs["publicNetworkAccessEnabled"] = args ? args.publicNetworkAccessEnabled : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["roleBasedAccessControl"] = args ? args.roleBasedAccessControl : undefined;
            inputs["servicePrincipal"] = args ? args.servicePrincipal : undefined;
            inputs["skuTier"] = args ? args.skuTier : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["windowsProfile"] = args ? args.windowsProfile : undefined;
            inputs["fqdn"] = undefined /*out*/;
            inputs["kubeAdminConfigRaw"] = undefined /*out*/;
            inputs["kubeAdminConfigs"] = undefined /*out*/;
            inputs["kubeConfigRaw"] = undefined /*out*/;
            inputs["kubeConfigs"] = undefined /*out*/;
            inputs["portalFqdn"] = undefined /*out*/;
            inputs["privateFqdn"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(KubernetesCluster.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing KubernetesCluster resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new KubernetesCluster(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of KubernetesCluster.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === KubernetesCluster.__pulumiType;
    }
}
exports.KubernetesCluster = KubernetesCluster;
/** @internal */
KubernetesCluster.__pulumiType = 'azure:containerservice/kubernetesCluster:KubernetesCluster';
