// @pulumi/azure/appinsights/insights.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Insights = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an Application Insights component.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const exampleResourceGroup = new azure.core.ResourceGroup("exampleResourceGroup", {location: "West Europe"});
 * const exampleInsights = new azure.appinsights.Insights("exampleInsights", {
 *     location: exampleResourceGroup.location,
 *     resourceGroupName: exampleResourceGroup.name,
 *     applicationType: "web",
 * });
 * export const instrumentationKey = exampleInsights.instrumentationKey;
 * export const appId = exampleInsights.appId;
 * ```
 * ### Workspace Mode
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const exampleResourceGroup = new azure.core.ResourceGroup("exampleResourceGroup", {location: "West Europe"});
 * const exampleAnalyticsWorkspace = new azure.operationalinsights.AnalyticsWorkspace("exampleAnalyticsWorkspace", {
 *     location: exampleResourceGroup.location,
 *     resourceGroupName: exampleResourceGroup.name,
 *     sku: "PerGB2018",
 *     retentionInDays: 30,
 * });
 * const exampleInsights = new azure.appinsights.Insights("exampleInsights", {
 *     location: exampleResourceGroup.location,
 *     resourceGroupName: exampleResourceGroup.name,
 *     workspaceId: exampleAnalyticsWorkspace.id,
 *     applicationType: "web",
 * });
 * export const instrumentationKey = exampleInsights.instrumentationKey;
 * export const appId = exampleInsights.appId;
 * ```
 *
 * ## Import
 *
 * Application Insights instances can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:appinsights/insights:Insights instance1 /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/Microsoft.Insights/components/instance1
 * ```
 */
class Insights extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            inputs["appId"] = state ? state.appId : undefined;
            inputs["applicationType"] = state ? state.applicationType : undefined;
            inputs["connectionString"] = state ? state.connectionString : undefined;
            inputs["dailyDataCapInGb"] = state ? state.dailyDataCapInGb : undefined;
            inputs["dailyDataCapNotificationsDisabled"] = state ? state.dailyDataCapNotificationsDisabled : undefined;
            inputs["disableIpMasking"] = state ? state.disableIpMasking : undefined;
            inputs["instrumentationKey"] = state ? state.instrumentationKey : undefined;
            inputs["internetIngestionEnabled"] = state ? state.internetIngestionEnabled : undefined;
            inputs["internetQueryEnabled"] = state ? state.internetQueryEnabled : undefined;
            inputs["localAuthenticationDisabled"] = state ? state.localAuthenticationDisabled : undefined;
            inputs["location"] = state ? state.location : undefined;
            inputs["name"] = state ? state.name : undefined;
            inputs["resourceGroupName"] = state ? state.resourceGroupName : undefined;
            inputs["retentionInDays"] = state ? state.retentionInDays : undefined;
            inputs["samplingPercentage"] = state ? state.samplingPercentage : undefined;
            inputs["tags"] = state ? state.tags : undefined;
            inputs["workspaceId"] = state ? state.workspaceId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.applicationType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'applicationType'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["applicationType"] = args ? args.applicationType : undefined;
            inputs["dailyDataCapInGb"] = args ? args.dailyDataCapInGb : undefined;
            inputs["dailyDataCapNotificationsDisabled"] = args ? args.dailyDataCapNotificationsDisabled : undefined;
            inputs["disableIpMasking"] = args ? args.disableIpMasking : undefined;
            inputs["internetIngestionEnabled"] = args ? args.internetIngestionEnabled : undefined;
            inputs["internetQueryEnabled"] = args ? args.internetQueryEnabled : undefined;
            inputs["localAuthenticationDisabled"] = args ? args.localAuthenticationDisabled : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["retentionInDays"] = args ? args.retentionInDays : undefined;
            inputs["samplingPercentage"] = args ? args.samplingPercentage : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["workspaceId"] = args ? args.workspaceId : undefined;
            inputs["appId"] = undefined /*out*/;
            inputs["connectionString"] = undefined /*out*/;
            inputs["instrumentationKey"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Insights.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Insights resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Insights(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Insights.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Insights.__pulumiType;
    }
}
exports.Insights = Insights;
/** @internal */
Insights.__pulumiType = 'azure:appinsights/insights:Insights';
