// @azure/eventgrid/dist-esm/src/generated/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
export const contentType = {
    parameterPath: ["options", "contentType"],
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Content-Type",
        type: {
            name: "String"
        }
    }
};
export const events = {
    parameterPath: "events",
    mapper: {
        serializedName: "events",
        required: true,
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "Composite",
                    className: "EventGridEvent"
                }
            }
        }
    }
};
export const topicHostname = {
    parameterPath: "topicHostname",
    mapper: {
        serializedName: "topicHostname",
        required: true,
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export const apiVersion = {
    parameterPath: "apiVersion",
    mapper: {
        defaultValue: "2018-01-01",
        isConstant: true,
        serializedName: "api-version",
        type: {
            name: "String"
        }
    }
};
export const contentType1 = {
    parameterPath: ["options", "contentType"],
    mapper: {
        defaultValue: "application/cloudevents-batch+json; charset=utf-8",
        isConstant: true,
        serializedName: "Content-Type",
        type: {
            name: "String"
        }
    }
};
export const events1 = {
    parameterPath: "events",
    mapper: {
        serializedName: "events",
        required: true,
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "Composite",
                    className: "CloudEvent"
                }
            }
        }
    }
};
export const aegChannelName = {
    parameterPath: ["options", "aegChannelName"],
    mapper: {
        serializedName: "aeg-channel-name",
        type: {
            name: "String"
        }
    }
};
export const events2 = {
    parameterPath: "events",
    mapper: {
        serializedName: "events",
        required: true,
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "any"
                }
            }
        }
    }
};
