// @pulumi/azure/monitoring/logzMonitor.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.LogzMonitor = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a logz Monitor.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const exampleResourceGroup = new azure.core.ResourceGroup("exampleResourceGroup", {location: "West Europe"});
 * const exampleLogzMonitor = new azure.monitoring.LogzMonitor("exampleLogzMonitor", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     location: exampleResourceGroup.location,
 *     plan: {
 *         billingCycle: "Monthly",
 *         effectiveDate: "2022-06-06T00:00:00Z",
 *         planId: "100gb14days",
 *         usageType: "Committed",
 *     },
 *     user: {
 *         email: "user@example.com",
 *         firstName: "Example",
 *         lastName: "User",
 *         phoneNumber: "+12313803556",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * logz Monitors can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:monitoring/logzMonitor:LogzMonitor example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Logz/monitors/monitor1
 * ```
 */
class LogzMonitor extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            inputs["companyName"] = state ? state.companyName : undefined;
            inputs["enabled"] = state ? state.enabled : undefined;
            inputs["enterpriseAppId"] = state ? state.enterpriseAppId : undefined;
            inputs["location"] = state ? state.location : undefined;
            inputs["logzOrganizationId"] = state ? state.logzOrganizationId : undefined;
            inputs["name"] = state ? state.name : undefined;
            inputs["plan"] = state ? state.plan : undefined;
            inputs["resourceGroupName"] = state ? state.resourceGroupName : undefined;
            inputs["singleSignOnUrl"] = state ? state.singleSignOnUrl : undefined;
            inputs["tags"] = state ? state.tags : undefined;
            inputs["user"] = state ? state.user : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.plan === undefined) && !opts.urn) {
                throw new Error("Missing required property 'plan'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.user === undefined) && !opts.urn) {
                throw new Error("Missing required property 'user'");
            }
            inputs["companyName"] = args ? args.companyName : undefined;
            inputs["enabled"] = args ? args.enabled : undefined;
            inputs["enterpriseAppId"] = args ? args.enterpriseAppId : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["plan"] = args ? args.plan : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["user"] = args ? args.user : undefined;
            inputs["logzOrganizationId"] = undefined /*out*/;
            inputs["singleSignOnUrl"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(LogzMonitor.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing LogzMonitor resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new LogzMonitor(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of LogzMonitor.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LogzMonitor.__pulumiType;
    }
}
exports.LogzMonitor = LogzMonitor;
/** @internal */
LogzMonitor.__pulumiType = 'azure:monitoring/logzMonitor:LogzMonitor';
