// @pulumi/azure/core/resourceGroupCostManagementExport.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceGroupCostManagementExport = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Cost Management Export for a Resource Group.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const exampleResourceGroup = new azure.core.ResourceGroup("exampleResourceGroup", {location: "West Europe"});
 * const exampleAccount = new azure.storage.Account("exampleAccount", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     location: exampleResourceGroup.location,
 *     accountTier: "Standard",
 *     accountReplicationType: "LRS",
 * });
 * const exampleContainer = new azure.storage.Container("exampleContainer", {storageAccountName: azurerm_storage_account.test.name});
 * const exampleResourceGroupCostManagementExport = new azure.core.ResourceGroupCostManagementExport("exampleResourceGroupCostManagementExport", {
 *     resourceGroupId: exampleResourceGroup.id,
 *     recurrenceType: "Monthly",
 *     recurrencePeriodStartDate: "2020-08-18T00:00:00Z",
 *     recurrencePeriodEndDate: "2020-09-18T00:00:00Z",
 *     exportDataStorageLocation: {
 *         containerId: azurerm_storage_container.test.resource_manager_id,
 *         rootFolderPath: "/root/updated",
 *     },
 *     exportDataOptions: {
 *         type: "Usage",
 *         timeFrame: "WeekToDate",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Cost Management Export for a Resource Group can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:core/resourceGroupCostManagementExport:ResourceGroupCostManagementExport example /subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/resGroup1/providers/Microsoft.CostManagement/exports/export1
 * ```
 */
class ResourceGroupCostManagementExport extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["active"] = state ? state.active : undefined;
            resourceInputs["exportDataOptions"] = state ? state.exportDataOptions : undefined;
            resourceInputs["exportDataStorageLocation"] = state ? state.exportDataStorageLocation : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["recurrencePeriodEndDate"] = state ? state.recurrencePeriodEndDate : undefined;
            resourceInputs["recurrencePeriodStartDate"] = state ? state.recurrencePeriodStartDate : undefined;
            resourceInputs["recurrenceType"] = state ? state.recurrenceType : undefined;
            resourceInputs["resourceGroupId"] = state ? state.resourceGroupId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.exportDataOptions === undefined) && !opts.urn) {
                throw new Error("Missing required property 'exportDataOptions'");
            }
            if ((!args || args.exportDataStorageLocation === undefined) && !opts.urn) {
                throw new Error("Missing required property 'exportDataStorageLocation'");
            }
            if ((!args || args.recurrencePeriodEndDate === undefined) && !opts.urn) {
                throw new Error("Missing required property 'recurrencePeriodEndDate'");
            }
            if ((!args || args.recurrencePeriodStartDate === undefined) && !opts.urn) {
                throw new Error("Missing required property 'recurrencePeriodStartDate'");
            }
            if ((!args || args.recurrenceType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'recurrenceType'");
            }
            if ((!args || args.resourceGroupId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupId'");
            }
            resourceInputs["active"] = args ? args.active : undefined;
            resourceInputs["exportDataOptions"] = args ? args.exportDataOptions : undefined;
            resourceInputs["exportDataStorageLocation"] = args ? args.exportDataStorageLocation : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["recurrencePeriodEndDate"] = args ? args.recurrencePeriodEndDate : undefined;
            resourceInputs["recurrencePeriodStartDate"] = args ? args.recurrencePeriodStartDate : undefined;
            resourceInputs["recurrenceType"] = args ? args.recurrenceType : undefined;
            resourceInputs["resourceGroupId"] = args ? args.resourceGroupId : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ResourceGroupCostManagementExport.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing ResourceGroupCostManagementExport resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ResourceGroupCostManagementExport(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ResourceGroupCostManagementExport.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ResourceGroupCostManagementExport.__pulumiType;
    }
}
exports.ResourceGroupCostManagementExport = ResourceGroupCostManagementExport;
/** @internal */
ResourceGroupCostManagementExport.__pulumiType = 'azure:core/resourceGroupCostManagementExport:ResourceGroupCostManagementExport';
