// @pulumi/azure/synapse/workspaceKey.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspaceKey = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
class WorkspaceKey extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            inputs["active"] = state ? state.active : undefined;
            inputs["cusomterManagedKeyName"] = state ? state.cusomterManagedKeyName : undefined;
            inputs["customerManagedKeyName"] = state ? state.customerManagedKeyName : undefined;
            inputs["customerManagedKeyVersionlessId"] = state ? state.customerManagedKeyVersionlessId : undefined;
            inputs["synapseWorkspaceId"] = state ? state.synapseWorkspaceId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.active === undefined) && !opts.urn) {
                throw new Error("Missing required property 'active'");
            }
            if ((!args || args.synapseWorkspaceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'synapseWorkspaceId'");
            }
            inputs["active"] = args ? args.active : undefined;
            inputs["cusomterManagedKeyName"] = args ? args.cusomterManagedKeyName : undefined;
            inputs["customerManagedKeyName"] = args ? args.customerManagedKeyName : undefined;
            inputs["customerManagedKeyVersionlessId"] = args ? args.customerManagedKeyVersionlessId : undefined;
            inputs["synapseWorkspaceId"] = args ? args.synapseWorkspaceId : undefined;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(WorkspaceKey.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing WorkspaceKey resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new WorkspaceKey(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WorkspaceKey.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WorkspaceKey.__pulumiType;
    }
}
exports.WorkspaceKey = WorkspaceKey;
/** @internal */
WorkspaceKey.__pulumiType = 'azure:synapse/workspaceKey:WorkspaceKey';
