// @pulumi/azure/provider.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Provider = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * The provider type for the azurerm package. By default, resources use package-wide configuration
 * settings, however an explicit `Provider` instance may be created and passed during resource
 * construction to achieve fine-grained programmatic control over provider settings. See the
 * [documentation](https://www.pulumi.com/docs/reference/programming-model/#providers) for more information.
 */
class Provider extends pulumi.ProviderResource {
    /**
     * Create a Provider resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a, _b, _c, _d, _e, _f;
        let resourceInputs = {};
        opts = opts || {};
        {
            resourceInputs["auxiliaryTenantIds"] = pulumi.output(args ? args.auxiliaryTenantIds : undefined).apply(JSON.stringify);
            resourceInputs["clientCertificatePassword"] = args ? args.clientCertificatePassword : undefined;
            resourceInputs["clientCertificatePath"] = args ? args.clientCertificatePath : undefined;
            resourceInputs["clientId"] = args ? args.clientId : undefined;
            resourceInputs["clientSecret"] = args ? args.clientSecret : undefined;
            resourceInputs["disableCorrelationRequestId"] = pulumi.output(args ? args.disableCorrelationRequestId : undefined).apply(JSON.stringify);
            resourceInputs["disableTerraformPartnerId"] = pulumi.output(args ? args.disableTerraformPartnerId : undefined).apply(JSON.stringify);
            resourceInputs["environment"] = (_a = (args ? args.environment : undefined)) !== null && _a !== void 0 ? _a : (utilities.getEnv("AZURE_ENVIRONMENT", "ARM_ENVIRONMENT") || "public");
            resourceInputs["features"] = pulumi.output(args ? args.features : undefined).apply(JSON.stringify);
            resourceInputs["metadataHost"] = (_b = (args ? args.metadataHost : undefined)) !== null && _b !== void 0 ? _b : utilities.getEnv("ARM_METADATA_HOSTNAME");
            resourceInputs["metadataUrl"] = (_c = (args ? args.metadataUrl : undefined)) !== null && _c !== void 0 ? _c : utilities.getEnv("ARM_METADATA_URL");
            resourceInputs["msiEndpoint"] = args ? args.msiEndpoint : undefined;
            resourceInputs["partnerId"] = args ? args.partnerId : undefined;
            resourceInputs["skipCredentialsValidation"] = pulumi.output(args ? args.skipCredentialsValidation : undefined).apply(JSON.stringify);
            resourceInputs["skipProviderRegistration"] = pulumi.output((_d = (args ? args.skipProviderRegistration : undefined)) !== null && _d !== void 0 ? _d : (utilities.getEnvBoolean("ARM_SKIP_PROVIDER_REGISTRATION") || false)).apply(JSON.stringify);
            resourceInputs["storageUseAzuread"] = pulumi.output((_e = (args ? args.storageUseAzuread : undefined)) !== null && _e !== void 0 ? _e : (utilities.getEnvBoolean("ARM_STORAGE_USE_AZUREAD") || false)).apply(JSON.stringify);
            resourceInputs["subscriptionId"] = (_f = (args ? args.subscriptionId : undefined)) !== null && _f !== void 0 ? _f : (utilities.getEnv("ARM_SUBSCRIPTION_ID") || "");
            resourceInputs["tenantId"] = args ? args.tenantId : undefined;
            resourceInputs["useMsal"] = pulumi.output(args ? args.useMsal : undefined).apply(JSON.stringify);
            resourceInputs["useMsi"] = pulumi.output(args ? args.useMsi : undefined).apply(JSON.stringify);
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Provider.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Returns true if the given object is an instance of Provider.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Provider.__pulumiType;
    }
}
exports.Provider = Provider;
/** @internal */
Provider.__pulumiType = 'azure';
