// @pulumi/azure/core/tenantTemplateDeployment.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TenantTemplateDeployment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const exampleTemplateSpecVersion = azure.core.getTemplateSpecVersion({
 *     name: "myTemplateForTenant",
 *     resourceGroupName: "myResourceGroup",
 *     version: "v0.1",
 * });
 * const exampleTenantTemplateDeployment = new azure.core.TenantTemplateDeployment("exampleTenantTemplateDeployment", {
 *     location: "West Europe",
 *     templateSpecVersionId: exampleTemplateSpecVersion.then(exampleTemplateSpecVersion => exampleTemplateSpecVersion.id),
 * });
 * ```
 *
 * ## Import
 *
 * Tenant Template Deployments can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:core/tenantTemplateDeployment:TenantTemplateDeployment example /providers/Microsoft.Resources/deployments/deploy1
 * ```
 */
class TenantTemplateDeployment extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["debugLevel"] = state ? state.debugLevel : undefined;
            resourceInputs["location"] = state ? state.location : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["outputContent"] = state ? state.outputContent : undefined;
            resourceInputs["parametersContent"] = state ? state.parametersContent : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
            resourceInputs["templateContent"] = state ? state.templateContent : undefined;
            resourceInputs["templateSpecVersionId"] = state ? state.templateSpecVersionId : undefined;
        }
        else {
            const args = argsOrState;
            resourceInputs["debugLevel"] = args ? args.debugLevel : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["parametersContent"] = args ? args.parametersContent : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["templateContent"] = args ? args.templateContent : undefined;
            resourceInputs["templateSpecVersionId"] = args ? args.templateSpecVersionId : undefined;
            resourceInputs["outputContent"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(TenantTemplateDeployment.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing TenantTemplateDeployment resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new TenantTemplateDeployment(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of TenantTemplateDeployment.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TenantTemplateDeployment.__pulumiType;
    }
}
exports.TenantTemplateDeployment = TenantTemplateDeployment;
/** @internal */
TenantTemplateDeployment.__pulumiType = 'azure:core/tenantTemplateDeployment:TenantTemplateDeployment';
