// @pulumi/azure/frontdoor/frontdoor.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Frontdoor = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an Azure Front Door instance.
 *
 * Azure Front Door Service is Microsoft's highly available and scalable web application acceleration platform and global HTTP(s) load balancer. It provides built-in DDoS protection and application layer security and caching. Front Door enables you to build applications that maximize and automate high-availability and performance for your end-users. Use Front Door with Azure services including Web/Mobile Apps, Cloud Services and Virtual Machines – or combine it with on-premises services for hybrid deployments and smooth cloud migration.
 *
 * Below are some of the key scenarios that Azure Front Door Service addresses:
 * * Use Front Door to improve application scale and availability with instant multi-region failover
 * * Use Front Door to improve application performance with SSL offload and routing requests to the fastest available application backend.
 * * Use Front Door for application layer security and DDoS protection for your application.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const exampleResourceGroup = new azure.core.ResourceGroup("exampleResourceGroup", {location: "West Europe"});
 * const exampleFrontdoor = new azure.frontdoor.Frontdoor("exampleFrontdoor", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     enforceBackendPoolsCertificateNameCheck: false,
 *     routingRules: [{
 *         name: "exampleRoutingRule1",
 *         acceptedProtocols: [
 *             "Http",
 *             "Https",
 *         ],
 *         patternsToMatches: ["/*"],
 *         frontendEndpoints: ["exampleFrontendEndpoint1"],
 *         forwardingConfiguration: {
 *             forwardingProtocol: "MatchRequest",
 *             backendPoolName: "exampleBackendBing",
 *         },
 *     }],
 *     backendPoolLoadBalancings: [{
 *         name: "exampleLoadBalancingSettings1",
 *     }],
 *     backendPoolHealthProbes: [{
 *         name: "exampleHealthProbeSetting1",
 *     }],
 *     backendPools: [{
 *         name: "exampleBackendBing",
 *         backends: [{
 *             hostHeader: "www.bing.com",
 *             address: "www.bing.com",
 *             httpPort: 80,
 *             httpsPort: 443,
 *         }],
 *         loadBalancingName: "exampleLoadBalancingSettings1",
 *         healthProbeName: "exampleHealthProbeSetting1",
 *     }],
 *     frontendEndpoints: [{
 *         name: "exampleFrontendEndpoint1",
 *         hostName: "example-FrontDoor.azurefd.net",
 *     }],
 * });
 * ```
 *
 * ## Import
 *
 * Front Doors can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:frontdoor/frontdoor:Frontdoor example /subscriptions/00000000-0000-0000-0000-000000000000/resourcegroups/mygroup1/providers/Microsoft.Network/frontDoors/frontdoor1
 * ```
 */
class Frontdoor extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            inputs["backendPoolHealthProbes"] = state ? state.backendPoolHealthProbes : undefined;
            inputs["backendPoolHealthProbesMap"] = state ? state.backendPoolHealthProbesMap : undefined;
            inputs["backendPoolLoadBalancingSettingsMap"] = state ? state.backendPoolLoadBalancingSettingsMap : undefined;
            inputs["backendPoolLoadBalancings"] = state ? state.backendPoolLoadBalancings : undefined;
            inputs["backendPools"] = state ? state.backendPools : undefined;
            inputs["backendPoolsMap"] = state ? state.backendPoolsMap : undefined;
            inputs["backendPoolsSendReceiveTimeoutSeconds"] = state ? state.backendPoolsSendReceiveTimeoutSeconds : undefined;
            inputs["cname"] = state ? state.cname : undefined;
            inputs["enforceBackendPoolsCertificateNameCheck"] = state ? state.enforceBackendPoolsCertificateNameCheck : undefined;
            inputs["explicitResourceOrders"] = state ? state.explicitResourceOrders : undefined;
            inputs["friendlyName"] = state ? state.friendlyName : undefined;
            inputs["frontendEndpoints"] = state ? state.frontendEndpoints : undefined;
            inputs["frontendEndpointsMap"] = state ? state.frontendEndpointsMap : undefined;
            inputs["headerFrontdoorId"] = state ? state.headerFrontdoorId : undefined;
            inputs["loadBalancerEnabled"] = state ? state.loadBalancerEnabled : undefined;
            inputs["location"] = state ? state.location : undefined;
            inputs["name"] = state ? state.name : undefined;
            inputs["resourceGroupName"] = state ? state.resourceGroupName : undefined;
            inputs["routingRules"] = state ? state.routingRules : undefined;
            inputs["routingRulesMap"] = state ? state.routingRulesMap : undefined;
            inputs["tags"] = state ? state.tags : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.backendPoolHealthProbes === undefined) && !opts.urn) {
                throw new Error("Missing required property 'backendPoolHealthProbes'");
            }
            if ((!args || args.backendPoolLoadBalancings === undefined) && !opts.urn) {
                throw new Error("Missing required property 'backendPoolLoadBalancings'");
            }
            if ((!args || args.backendPools === undefined) && !opts.urn) {
                throw new Error("Missing required property 'backendPools'");
            }
            if ((!args || args.enforceBackendPoolsCertificateNameCheck === undefined) && !opts.urn) {
                throw new Error("Missing required property 'enforceBackendPoolsCertificateNameCheck'");
            }
            if ((!args || args.frontendEndpoints === undefined) && !opts.urn) {
                throw new Error("Missing required property 'frontendEndpoints'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.routingRules === undefined) && !opts.urn) {
                throw new Error("Missing required property 'routingRules'");
            }
            inputs["backendPoolHealthProbes"] = args ? args.backendPoolHealthProbes : undefined;
            inputs["backendPoolLoadBalancings"] = args ? args.backendPoolLoadBalancings : undefined;
            inputs["backendPools"] = args ? args.backendPools : undefined;
            inputs["backendPoolsSendReceiveTimeoutSeconds"] = args ? args.backendPoolsSendReceiveTimeoutSeconds : undefined;
            inputs["enforceBackendPoolsCertificateNameCheck"] = args ? args.enforceBackendPoolsCertificateNameCheck : undefined;
            inputs["friendlyName"] = args ? args.friendlyName : undefined;
            inputs["frontendEndpoints"] = args ? args.frontendEndpoints : undefined;
            inputs["loadBalancerEnabled"] = args ? args.loadBalancerEnabled : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["routingRules"] = args ? args.routingRules : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["backendPoolHealthProbesMap"] = undefined /*out*/;
            inputs["backendPoolLoadBalancingSettingsMap"] = undefined /*out*/;
            inputs["backendPoolsMap"] = undefined /*out*/;
            inputs["cname"] = undefined /*out*/;
            inputs["explicitResourceOrders"] = undefined /*out*/;
            inputs["frontendEndpointsMap"] = undefined /*out*/;
            inputs["headerFrontdoorId"] = undefined /*out*/;
            inputs["routingRulesMap"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Frontdoor.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Frontdoor resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Frontdoor(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Frontdoor.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Frontdoor.__pulumiType;
    }
}
exports.Frontdoor = Frontdoor;
/** @internal */
Frontdoor.__pulumiType = 'azure:frontdoor/frontdoor:Frontdoor';
