// @pulumi/azure/iot/timeSeriesInsightsEventSourceIothub.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimeSeriesInsightsEventSourceIothub = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an Azure IoT Time Series Insights IoTHub Event Source.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const exampleResourceGroup = new azure.core.ResourceGroup("exampleResourceGroup", {location: "West Europe"});
 * const exampleIoTHub = new azure.iot.IoTHub("exampleIoTHub", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     location: exampleResourceGroup.location,
 *     sku: {
 *         name: "B1",
 *         capacity: "1",
 *     },
 * });
 * const exampleConsumerGroup = new azure.iot.ConsumerGroup("exampleConsumerGroup", {
 *     iothubName: exampleIoTHub.name,
 *     eventhubEndpointName: "events",
 *     resourceGroupName: exampleResourceGroup.name,
 * });
 * const storage = new azure.storage.Account("storage", {
 *     location: exampleResourceGroup.location,
 *     resourceGroupName: exampleResourceGroup.name,
 *     accountTier: "Standard",
 *     accountReplicationType: "LRS",
 * });
 * const exampleTimeSeriesInsightsGen2Environment = new azure.iot.TimeSeriesInsightsGen2Environment("exampleTimeSeriesInsightsGen2Environment", {
 *     location: exampleResourceGroup.location,
 *     resourceGroupName: exampleResourceGroup.name,
 *     skuName: "L1",
 *     idProperties: ["id"],
 *     storage: {
 *         name: storage.name,
 *         key: storage.primaryAccessKey,
 *     },
 * });
 * const exampleTimeSeriesInsightsEventSourceIothub = new azure.iot.TimeSeriesInsightsEventSourceIothub("exampleTimeSeriesInsightsEventSourceIothub", {
 *     location: exampleResourceGroup.location,
 *     environmentId: exampleTimeSeriesInsightsGen2Environment.id,
 *     iothubName: exampleIoTHub.name,
 *     sharedAccessKey: exampleIoTHub.sharedAccessPolicies.apply(sharedAccessPolicies => sharedAccessPolicies[0].primaryKey),
 *     sharedAccessKeyName: exampleIoTHub.sharedAccessPolicies.apply(sharedAccessPolicies => sharedAccessPolicies[0].keyName),
 *     consumerGroupName: exampleConsumerGroup.name,
 *     eventSourceResourceId: exampleIoTHub.id,
 * });
 * ```
 *
 * ## Import
 *
 * Azure IoT Time Series Insights IoTHub Event Source can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:iot/timeSeriesInsightsEventSourceIothub:TimeSeriesInsightsEventSourceIothub example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/example/providers/Microsoft.TimeSeriesInsights/environments/environment1/eventSources/example
 * ```
 */
class TimeSeriesInsightsEventSourceIothub extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            inputs["consumerGroupName"] = state ? state.consumerGroupName : undefined;
            inputs["environmentId"] = state ? state.environmentId : undefined;
            inputs["eventSourceResourceId"] = state ? state.eventSourceResourceId : undefined;
            inputs["iothubName"] = state ? state.iothubName : undefined;
            inputs["location"] = state ? state.location : undefined;
            inputs["name"] = state ? state.name : undefined;
            inputs["sharedAccessKey"] = state ? state.sharedAccessKey : undefined;
            inputs["sharedAccessKeyName"] = state ? state.sharedAccessKeyName : undefined;
            inputs["tags"] = state ? state.tags : undefined;
            inputs["timestampPropertyName"] = state ? state.timestampPropertyName : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.consumerGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'consumerGroupName'");
            }
            if ((!args || args.environmentId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'environmentId'");
            }
            if ((!args || args.eventSourceResourceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'eventSourceResourceId'");
            }
            if ((!args || args.iothubName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'iothubName'");
            }
            if ((!args || args.sharedAccessKey === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sharedAccessKey'");
            }
            if ((!args || args.sharedAccessKeyName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sharedAccessKeyName'");
            }
            inputs["consumerGroupName"] = args ? args.consumerGroupName : undefined;
            inputs["environmentId"] = args ? args.environmentId : undefined;
            inputs["eventSourceResourceId"] = args ? args.eventSourceResourceId : undefined;
            inputs["iothubName"] = args ? args.iothubName : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["sharedAccessKey"] = args ? args.sharedAccessKey : undefined;
            inputs["sharedAccessKeyName"] = args ? args.sharedAccessKeyName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["timestampPropertyName"] = args ? args.timestampPropertyName : undefined;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(TimeSeriesInsightsEventSourceIothub.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing TimeSeriesInsightsEventSourceIothub resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new TimeSeriesInsightsEventSourceIothub(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of TimeSeriesInsightsEventSourceIothub.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TimeSeriesInsightsEventSourceIothub.__pulumiType;
    }
}
exports.TimeSeriesInsightsEventSourceIothub = TimeSeriesInsightsEventSourceIothub;
/** @internal */
TimeSeriesInsightsEventSourceIothub.__pulumiType = 'azure:iot/timeSeriesInsightsEventSourceIothub:TimeSeriesInsightsEventSourceIothub';
