// @pulumi/azure/network/applicationGateway.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApplicationGateway = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an Application Gateway.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const exampleResourceGroup = new azure.core.ResourceGroup("exampleResourceGroup", {location: "West Europe"});
 * const exampleVirtualNetwork = new azure.network.VirtualNetwork("exampleVirtualNetwork", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     location: exampleResourceGroup.location,
 *     addressSpaces: ["10.254.0.0/16"],
 * });
 * const frontend = new azure.network.Subnet("frontend", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     virtualNetworkName: exampleVirtualNetwork.name,
 *     addressPrefixes: ["10.254.0.0/24"],
 * });
 * const backend = new azure.network.Subnet("backend", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     virtualNetworkName: exampleVirtualNetwork.name,
 *     addressPrefixes: ["10.254.2.0/24"],
 * });
 * const examplePublicIp = new azure.network.PublicIp("examplePublicIp", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     location: exampleResourceGroup.location,
 *     allocationMethod: "Dynamic",
 * });
 * const backendAddressPoolName = pulumi.interpolate`${exampleVirtualNetwork.name}-beap`;
 * const frontendPortName = pulumi.interpolate`${exampleVirtualNetwork.name}-feport`;
 * const frontendIpConfigurationName = pulumi.interpolate`${exampleVirtualNetwork.name}-feip`;
 * const httpSettingName = pulumi.interpolate`${exampleVirtualNetwork.name}-be-htst`;
 * const listenerName = pulumi.interpolate`${exampleVirtualNetwork.name}-httplstn`;
 * const requestRoutingRuleName = pulumi.interpolate`${exampleVirtualNetwork.name}-rqrt`;
 * const redirectConfigurationName = pulumi.interpolate`${exampleVirtualNetwork.name}-rdrcfg`;
 * const network = new azure.network.ApplicationGateway("network", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     location: exampleResourceGroup.location,
 *     sku: {
 *         name: "Standard_Small",
 *         tier: "Standard",
 *         capacity: 2,
 *     },
 *     gatewayIpConfigurations: [{
 *         name: "my-gateway-ip-configuration",
 *         subnetId: frontend.id,
 *     }],
 *     frontendPorts: [{
 *         name: frontendPortName,
 *         port: 80,
 *     }],
 *     frontendIpConfigurations: [{
 *         name: frontendIpConfigurationName,
 *         publicIpAddressId: examplePublicIp.id,
 *     }],
 *     backendAddressPools: [{
 *         name: backendAddressPoolName,
 *     }],
 *     backendHttpSettings: [{
 *         name: httpSettingName,
 *         cookieBasedAffinity: "Disabled",
 *         path: "/path1/",
 *         port: 80,
 *         protocol: "Http",
 *         requestTimeout: 60,
 *     }],
 *     httpListeners: [{
 *         name: listenerName,
 *         frontendIpConfigurationName: frontendIpConfigurationName,
 *         frontendPortName: frontendPortName,
 *         protocol: "Http",
 *     }],
 *     requestRoutingRules: [{
 *         name: requestRoutingRuleName,
 *         ruleType: "Basic",
 *         httpListenerName: listenerName,
 *         backendAddressPoolName: backendAddressPoolName,
 *         backendHttpSettingsName: httpSettingName,
 *     }],
 * });
 * ```
 *
 * ## Import
 *
 * Application Gateway's can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:network/applicationGateway:ApplicationGateway example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/Microsoft.Network/applicationGateways/myGateway1
 * ```
 */
class ApplicationGateway extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["authenticationCertificates"] = state ? state.authenticationCertificates : undefined;
            resourceInputs["autoscaleConfiguration"] = state ? state.autoscaleConfiguration : undefined;
            resourceInputs["backendAddressPools"] = state ? state.backendAddressPools : undefined;
            resourceInputs["backendHttpSettings"] = state ? state.backendHttpSettings : undefined;
            resourceInputs["customErrorConfigurations"] = state ? state.customErrorConfigurations : undefined;
            resourceInputs["enableHttp2"] = state ? state.enableHttp2 : undefined;
            resourceInputs["fipsEnabled"] = state ? state.fipsEnabled : undefined;
            resourceInputs["firewallPolicyId"] = state ? state.firewallPolicyId : undefined;
            resourceInputs["forceFirewallPolicyAssociation"] = state ? state.forceFirewallPolicyAssociation : undefined;
            resourceInputs["frontendIpConfigurations"] = state ? state.frontendIpConfigurations : undefined;
            resourceInputs["frontendPorts"] = state ? state.frontendPorts : undefined;
            resourceInputs["gatewayIpConfigurations"] = state ? state.gatewayIpConfigurations : undefined;
            resourceInputs["httpListeners"] = state ? state.httpListeners : undefined;
            resourceInputs["identity"] = state ? state.identity : undefined;
            resourceInputs["location"] = state ? state.location : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["privateEndpointConnections"] = state ? state.privateEndpointConnections : undefined;
            resourceInputs["privateLinkConfigurations"] = state ? state.privateLinkConfigurations : undefined;
            resourceInputs["probes"] = state ? state.probes : undefined;
            resourceInputs["redirectConfigurations"] = state ? state.redirectConfigurations : undefined;
            resourceInputs["requestRoutingRules"] = state ? state.requestRoutingRules : undefined;
            resourceInputs["resourceGroupName"] = state ? state.resourceGroupName : undefined;
            resourceInputs["rewriteRuleSets"] = state ? state.rewriteRuleSets : undefined;
            resourceInputs["sku"] = state ? state.sku : undefined;
            resourceInputs["sslCertificates"] = state ? state.sslCertificates : undefined;
            resourceInputs["sslPolicies"] = state ? state.sslPolicies : undefined;
            resourceInputs["sslProfiles"] = state ? state.sslProfiles : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
            resourceInputs["trustedClientCertificates"] = state ? state.trustedClientCertificates : undefined;
            resourceInputs["trustedRootCertificates"] = state ? state.trustedRootCertificates : undefined;
            resourceInputs["urlPathMaps"] = state ? state.urlPathMaps : undefined;
            resourceInputs["wafConfiguration"] = state ? state.wafConfiguration : undefined;
            resourceInputs["zones"] = state ? state.zones : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.backendAddressPools === undefined) && !opts.urn) {
                throw new Error("Missing required property 'backendAddressPools'");
            }
            if ((!args || args.backendHttpSettings === undefined) && !opts.urn) {
                throw new Error("Missing required property 'backendHttpSettings'");
            }
            if ((!args || args.frontendIpConfigurations === undefined) && !opts.urn) {
                throw new Error("Missing required property 'frontendIpConfigurations'");
            }
            if ((!args || args.frontendPorts === undefined) && !opts.urn) {
                throw new Error("Missing required property 'frontendPorts'");
            }
            if ((!args || args.gatewayIpConfigurations === undefined) && !opts.urn) {
                throw new Error("Missing required property 'gatewayIpConfigurations'");
            }
            if ((!args || args.httpListeners === undefined) && !opts.urn) {
                throw new Error("Missing required property 'httpListeners'");
            }
            if ((!args || args.requestRoutingRules === undefined) && !opts.urn) {
                throw new Error("Missing required property 'requestRoutingRules'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.sku === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sku'");
            }
            resourceInputs["authenticationCertificates"] = args ? args.authenticationCertificates : undefined;
            resourceInputs["autoscaleConfiguration"] = args ? args.autoscaleConfiguration : undefined;
            resourceInputs["backendAddressPools"] = args ? args.backendAddressPools : undefined;
            resourceInputs["backendHttpSettings"] = args ? args.backendHttpSettings : undefined;
            resourceInputs["customErrorConfigurations"] = args ? args.customErrorConfigurations : undefined;
            resourceInputs["enableHttp2"] = args ? args.enableHttp2 : undefined;
            resourceInputs["fipsEnabled"] = args ? args.fipsEnabled : undefined;
            resourceInputs["firewallPolicyId"] = args ? args.firewallPolicyId : undefined;
            resourceInputs["forceFirewallPolicyAssociation"] = args ? args.forceFirewallPolicyAssociation : undefined;
            resourceInputs["frontendIpConfigurations"] = args ? args.frontendIpConfigurations : undefined;
            resourceInputs["frontendPorts"] = args ? args.frontendPorts : undefined;
            resourceInputs["gatewayIpConfigurations"] = args ? args.gatewayIpConfigurations : undefined;
            resourceInputs["httpListeners"] = args ? args.httpListeners : undefined;
            resourceInputs["identity"] = args ? args.identity : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["privateLinkConfigurations"] = args ? args.privateLinkConfigurations : undefined;
            resourceInputs["probes"] = args ? args.probes : undefined;
            resourceInputs["redirectConfigurations"] = args ? args.redirectConfigurations : undefined;
            resourceInputs["requestRoutingRules"] = args ? args.requestRoutingRules : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["rewriteRuleSets"] = args ? args.rewriteRuleSets : undefined;
            resourceInputs["sku"] = args ? args.sku : undefined;
            resourceInputs["sslCertificates"] = args ? args.sslCertificates : undefined;
            resourceInputs["sslPolicies"] = args ? args.sslPolicies : undefined;
            resourceInputs["sslProfiles"] = args ? args.sslProfiles : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["trustedClientCertificates"] = args ? args.trustedClientCertificates : undefined;
            resourceInputs["trustedRootCertificates"] = args ? args.trustedRootCertificates : undefined;
            resourceInputs["urlPathMaps"] = args ? args.urlPathMaps : undefined;
            resourceInputs["wafConfiguration"] = args ? args.wafConfiguration : undefined;
            resourceInputs["zones"] = args ? args.zones : undefined;
            resourceInputs["privateEndpointConnections"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ApplicationGateway.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing ApplicationGateway resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ApplicationGateway(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ApplicationGateway.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ApplicationGateway.__pulumiType;
    }
}
exports.ApplicationGateway = ApplicationGateway;
/** @internal */
ApplicationGateway.__pulumiType = 'azure:network/applicationGateway:ApplicationGateway';
