// @pulumi/azure/appplatform/springCloudCustomDomain.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpringCloudCustomDomain = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an Azure Spring Cloud Custom Domain.
 *
 * ## Import
 *
 * Spring Cloud Custom Domain can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:appplatform/springCloudCustomDomain:SpringCloudCustomDomain example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resGroup1/providers/Microsoft.AppPlatform/Spring/spring1/apps/app1/domains/domain.com
 * ```
 */
class SpringCloudCustomDomain extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            inputs["certificateName"] = state ? state.certificateName : undefined;
            inputs["name"] = state ? state.name : undefined;
            inputs["springCloudAppId"] = state ? state.springCloudAppId : undefined;
            inputs["thumbprint"] = state ? state.thumbprint : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.springCloudAppId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'springCloudAppId'");
            }
            inputs["certificateName"] = args ? args.certificateName : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["springCloudAppId"] = args ? args.springCloudAppId : undefined;
            inputs["thumbprint"] = args ? args.thumbprint : undefined;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(SpringCloudCustomDomain.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing SpringCloudCustomDomain resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new SpringCloudCustomDomain(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SpringCloudCustomDomain.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SpringCloudCustomDomain.__pulumiType;
    }
}
exports.SpringCloudCustomDomain = SpringCloudCustomDomain;
/** @internal */
SpringCloudCustomDomain.__pulumiType = 'azure:appplatform/springCloudCustomDomain:SpringCloudCustomDomain';
