// @pulumi/azure/network/localNetworkGateway.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.LocalNetworkGateway = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a local network gateway connection over which specific connections can be configured.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = new azure.core.ResourceGroup("example", {location: "West Europe"});
 * const home = new azure.network.LocalNetworkGateway("home", {
 *     resourceGroupName: example.name,
 *     location: example.location,
 *     gatewayAddress: "12.13.14.15",
 *     addressSpaces: ["10.0.0.0/16"],
 * });
 * ```
 *
 * ## Import
 *
 * Local Network Gateways can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:network/localNetworkGateway:LocalNetworkGateway lng1 /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/Microsoft.Network/localNetworkGateways/lng1
 * ```
 */
class LocalNetworkGateway extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            inputs["addressSpaces"] = state ? state.addressSpaces : undefined;
            inputs["bgpSettings"] = state ? state.bgpSettings : undefined;
            inputs["gatewayAddress"] = state ? state.gatewayAddress : undefined;
            inputs["gatewayFqdn"] = state ? state.gatewayFqdn : undefined;
            inputs["location"] = state ? state.location : undefined;
            inputs["name"] = state ? state.name : undefined;
            inputs["resourceGroupName"] = state ? state.resourceGroupName : undefined;
            inputs["tags"] = state ? state.tags : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["addressSpaces"] = args ? args.addressSpaces : undefined;
            inputs["bgpSettings"] = args ? args.bgpSettings : undefined;
            inputs["gatewayAddress"] = args ? args.gatewayAddress : undefined;
            inputs["gatewayFqdn"] = args ? args.gatewayFqdn : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(LocalNetworkGateway.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing LocalNetworkGateway resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new LocalNetworkGateway(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of LocalNetworkGateway.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LocalNetworkGateway.__pulumiType;
    }
}
exports.LocalNetworkGateway = LocalNetworkGateway;
/** @internal */
LocalNetworkGateway.__pulumiType = 'azure:network/localNetworkGateway:LocalNetworkGateway';
