// @pulumi/azure/logicapps/triggerHttpRequest.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TriggerHttpRequest = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a HTTP Request Trigger within a Logic App Workflow
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const exampleResourceGroup = new azure.core.ResourceGroup("exampleResourceGroup", {location: "West Europe"});
 * const exampleWorkflow = new azure.logicapps.Workflow("exampleWorkflow", {
 *     location: exampleResourceGroup.location,
 *     resourceGroupName: exampleResourceGroup.name,
 * });
 * const exampleTriggerHttpRequest = new azure.logicapps.TriggerHttpRequest("exampleTriggerHttpRequest", {
 *     logicAppId: exampleWorkflow.id,
 *     schema: `{
 *     "type": "object",
 *     "properties": {
 *         "hello": {
 *             "type": "string"
 *         }
 *     }
 * }
 * `,
 * });
 * ```
 *
 * ## Import
 *
 * Logic App HTTP Request Triggers can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:logicapps/triggerHttpRequest:TriggerHttpRequest request1 /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/Microsoft.Logic/workflows/workflow1/triggers/request1
 * ```
 */
class TriggerHttpRequest extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["callbackUrl"] = state ? state.callbackUrl : undefined;
            resourceInputs["logicAppId"] = state ? state.logicAppId : undefined;
            resourceInputs["method"] = state ? state.method : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["relativePath"] = state ? state.relativePath : undefined;
            resourceInputs["schema"] = state ? state.schema : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.logicAppId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'logicAppId'");
            }
            if ((!args || args.schema === undefined) && !opts.urn) {
                throw new Error("Missing required property 'schema'");
            }
            resourceInputs["logicAppId"] = args ? args.logicAppId : undefined;
            resourceInputs["method"] = args ? args.method : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["relativePath"] = args ? args.relativePath : undefined;
            resourceInputs["schema"] = args ? args.schema : undefined;
            resourceInputs["callbackUrl"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(TriggerHttpRequest.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing TriggerHttpRequest resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new TriggerHttpRequest(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of TriggerHttpRequest.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TriggerHttpRequest.__pulumiType;
    }
}
exports.TriggerHttpRequest = TriggerHttpRequest;
/** @internal */
TriggerHttpRequest.__pulumiType = 'azure:logicapps/triggerHttpRequest:TriggerHttpRequest';
