// @pulumi/azure/bot/channelFacebook.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChannelFacebook = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Facebook integration for a Bot Channel
 *
 * > **Note** A bot can only have a single Facebook Channel associated with it.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const current = azure.core.getClientConfig({});
 * const exampleResourceGroup = new azure.core.ResourceGroup("exampleResourceGroup", {location: "West Europe"});
 * const exampleChannelsRegistration = new azure.bot.ChannelsRegistration("exampleChannelsRegistration", {
 *     location: "global",
 *     resourceGroupName: exampleResourceGroup.name,
 *     sku: "F0",
 *     microsoftAppId: current.then(current => current.clientId),
 * });
 * const exampleChannelFacebook = new azure.bot.ChannelFacebook("exampleChannelFacebook", {
 *     botName: exampleChannelsRegistration.name,
 *     location: exampleChannelsRegistration.location,
 *     resourceGroupName: exampleResourceGroup.name,
 *     facebookApplicationId: "563490254873576",
 *     facebookApplicationSecret: "8976d2536445ad5b976dee8437b9beb0",
 *     pages: [{
 *         id: "876248795081953",
 *         accessToken: "CGGCec3UAFPMBAKwK3Ft8SEpO8ZCuvpNBI5DClaJCDfqJj2BgEHCKxcY0FDarmUQap6XxpZC9GWCW4nZCzjcKosAZAP7SO44X8Q8gAntbDIXgYUBGp9xtS8wUkwgKPobUePcOOVFkvClxvYZByuiQxoTiK9fQ9jZCPEorbmZCsKDZAx4VLnrNwCTZAPUwXxO61gfq4ZD",
 *     }],
 * });
 * ```
 *
 * ## Import
 *
 * The Facebook Integration for a Bot Channel can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:bot/channelFacebook:ChannelFacebook example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.BotService/botServices/botService1/channels/FacebookChannel
 * ```
 */
class ChannelFacebook extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["botName"] = state ? state.botName : undefined;
            resourceInputs["facebookApplicationId"] = state ? state.facebookApplicationId : undefined;
            resourceInputs["facebookApplicationSecret"] = state ? state.facebookApplicationSecret : undefined;
            resourceInputs["location"] = state ? state.location : undefined;
            resourceInputs["pages"] = state ? state.pages : undefined;
            resourceInputs["resourceGroupName"] = state ? state.resourceGroupName : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.botName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'botName'");
            }
            if ((!args || args.facebookApplicationId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'facebookApplicationId'");
            }
            if ((!args || args.facebookApplicationSecret === undefined) && !opts.urn) {
                throw new Error("Missing required property 'facebookApplicationSecret'");
            }
            if ((!args || args.pages === undefined) && !opts.urn) {
                throw new Error("Missing required property 'pages'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["botName"] = args ? args.botName : undefined;
            resourceInputs["facebookApplicationId"] = args ? args.facebookApplicationId : undefined;
            resourceInputs["facebookApplicationSecret"] = args ? args.facebookApplicationSecret : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["pages"] = args ? args.pages : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ChannelFacebook.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing ChannelFacebook resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ChannelFacebook(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ChannelFacebook.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ChannelFacebook.__pulumiType;
    }
}
exports.ChannelFacebook = ChannelFacebook;
/** @internal */
ChannelFacebook.__pulumiType = 'azure:bot/channelFacebook:ChannelFacebook';
