// @pulumi/azure/kusto/cluster.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Cluster = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Kusto (also known as Azure Data Explorer) Cluster
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const rg = new azure.core.ResourceGroup("rg", {location: "West Europe"});
 * const example = new azure.kusto.Cluster("example", {
 *     location: rg.location,
 *     resourceGroupName: rg.name,
 *     sku: {
 *         name: "Standard_D13_v2",
 *         capacity: 2,
 *     },
 *     tags: {
 *         Environment: "Production",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Kusto Clusters can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:kusto/cluster:Cluster example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Kusto/Clusters/cluster1
 * ```
 */
class Cluster extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["autoStopEnabled"] = state ? state.autoStopEnabled : undefined;
            resourceInputs["dataIngestionUri"] = state ? state.dataIngestionUri : undefined;
            resourceInputs["diskEncryptionEnabled"] = state ? state.diskEncryptionEnabled : undefined;
            resourceInputs["doubleEncryptionEnabled"] = state ? state.doubleEncryptionEnabled : undefined;
            resourceInputs["enableAutoStop"] = state ? state.enableAutoStop : undefined;
            resourceInputs["enableDiskEncryption"] = state ? state.enableDiskEncryption : undefined;
            resourceInputs["enablePurge"] = state ? state.enablePurge : undefined;
            resourceInputs["enableStreamingIngest"] = state ? state.enableStreamingIngest : undefined;
            resourceInputs["engine"] = state ? state.engine : undefined;
            resourceInputs["identity"] = state ? state.identity : undefined;
            resourceInputs["languageExtensions"] = state ? state.languageExtensions : undefined;
            resourceInputs["location"] = state ? state.location : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["optimizedAutoScale"] = state ? state.optimizedAutoScale : undefined;
            resourceInputs["publicNetworkAccessEnabled"] = state ? state.publicNetworkAccessEnabled : undefined;
            resourceInputs["purgeEnabled"] = state ? state.purgeEnabled : undefined;
            resourceInputs["resourceGroupName"] = state ? state.resourceGroupName : undefined;
            resourceInputs["sku"] = state ? state.sku : undefined;
            resourceInputs["streamingIngestionEnabled"] = state ? state.streamingIngestionEnabled : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
            resourceInputs["trustedExternalTenants"] = state ? state.trustedExternalTenants : undefined;
            resourceInputs["uri"] = state ? state.uri : undefined;
            resourceInputs["virtualNetworkConfiguration"] = state ? state.virtualNetworkConfiguration : undefined;
            resourceInputs["zones"] = state ? state.zones : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.sku === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sku'");
            }
            resourceInputs["autoStopEnabled"] = args ? args.autoStopEnabled : undefined;
            resourceInputs["diskEncryptionEnabled"] = args ? args.diskEncryptionEnabled : undefined;
            resourceInputs["doubleEncryptionEnabled"] = args ? args.doubleEncryptionEnabled : undefined;
            resourceInputs["enableAutoStop"] = args ? args.enableAutoStop : undefined;
            resourceInputs["enableDiskEncryption"] = args ? args.enableDiskEncryption : undefined;
            resourceInputs["enablePurge"] = args ? args.enablePurge : undefined;
            resourceInputs["enableStreamingIngest"] = args ? args.enableStreamingIngest : undefined;
            resourceInputs["engine"] = args ? args.engine : undefined;
            resourceInputs["identity"] = args ? args.identity : undefined;
            resourceInputs["languageExtensions"] = args ? args.languageExtensions : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["optimizedAutoScale"] = args ? args.optimizedAutoScale : undefined;
            resourceInputs["publicNetworkAccessEnabled"] = args ? args.publicNetworkAccessEnabled : undefined;
            resourceInputs["purgeEnabled"] = args ? args.purgeEnabled : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["sku"] = args ? args.sku : undefined;
            resourceInputs["streamingIngestionEnabled"] = args ? args.streamingIngestionEnabled : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["trustedExternalTenants"] = args ? args.trustedExternalTenants : undefined;
            resourceInputs["virtualNetworkConfiguration"] = args ? args.virtualNetworkConfiguration : undefined;
            resourceInputs["zones"] = args ? args.zones : undefined;
            resourceInputs["dataIngestionUri"] = undefined /*out*/;
            resourceInputs["uri"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Cluster.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Cluster resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Cluster(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Cluster.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Cluster.__pulumiType;
    }
}
exports.Cluster = Cluster;
/** @internal */
Cluster.__pulumiType = 'azure:kusto/cluster:Cluster';
