// @pulumi/azure/containerservice/registryTask.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RegistryTask = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Container Registry Task.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const exampleResourceGroup = new azure.core.ResourceGroup("exampleResourceGroup", {location: "West Europe"});
 * const exampleRegistry = new azure.containerservice.Registry("exampleRegistry", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     location: exampleResourceGroup.location,
 *     sku: "Basic",
 * });
 * const exampleRegistryTask = new azure.containerservice.RegistryTask("exampleRegistryTask", {
 *     containerRegistryId: exampleRegistry.id,
 *     platform: {
 *         os: "Linux",
 *     },
 *     dockerStep: {
 *         dockerfilePath: "Dockerfile",
 *         contextPath: "https://github.com/<user name>/acr-build-helloworld-node#main",
 *         contextAccessToken: "<github personal access token>",
 *         imageNames: ["helloworld:{{.Run.ID}}"],
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Container Registry Tasks can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:containerservice/registryTask:RegistryTask example /subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/group1/providers/Microsoft.ContainerRegistry/registries/registry1/tasks/task1
 * ```
 */
class RegistryTask extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["agentPoolName"] = state ? state.agentPoolName : undefined;
            resourceInputs["agentSetting"] = state ? state.agentSetting : undefined;
            resourceInputs["baseImageTrigger"] = state ? state.baseImageTrigger : undefined;
            resourceInputs["containerRegistryId"] = state ? state.containerRegistryId : undefined;
            resourceInputs["dockerStep"] = state ? state.dockerStep : undefined;
            resourceInputs["enabled"] = state ? state.enabled : undefined;
            resourceInputs["encodedStep"] = state ? state.encodedStep : undefined;
            resourceInputs["fileStep"] = state ? state.fileStep : undefined;
            resourceInputs["identity"] = state ? state.identity : undefined;
            resourceInputs["isSystemTask"] = state ? state.isSystemTask : undefined;
            resourceInputs["logTemplate"] = state ? state.logTemplate : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["platform"] = state ? state.platform : undefined;
            resourceInputs["registryCredential"] = state ? state.registryCredential : undefined;
            resourceInputs["sourceTriggers"] = state ? state.sourceTriggers : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
            resourceInputs["timeoutInSeconds"] = state ? state.timeoutInSeconds : undefined;
            resourceInputs["timerTriggers"] = state ? state.timerTriggers : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.containerRegistryId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'containerRegistryId'");
            }
            resourceInputs["agentPoolName"] = args ? args.agentPoolName : undefined;
            resourceInputs["agentSetting"] = args ? args.agentSetting : undefined;
            resourceInputs["baseImageTrigger"] = args ? args.baseImageTrigger : undefined;
            resourceInputs["containerRegistryId"] = args ? args.containerRegistryId : undefined;
            resourceInputs["dockerStep"] = args ? args.dockerStep : undefined;
            resourceInputs["enabled"] = args ? args.enabled : undefined;
            resourceInputs["encodedStep"] = args ? args.encodedStep : undefined;
            resourceInputs["fileStep"] = args ? args.fileStep : undefined;
            resourceInputs["identity"] = args ? args.identity : undefined;
            resourceInputs["isSystemTask"] = args ? args.isSystemTask : undefined;
            resourceInputs["logTemplate"] = args ? args.logTemplate : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["platform"] = args ? args.platform : undefined;
            resourceInputs["registryCredential"] = args ? args.registryCredential : undefined;
            resourceInputs["sourceTriggers"] = args ? args.sourceTriggers : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["timeoutInSeconds"] = args ? args.timeoutInSeconds : undefined;
            resourceInputs["timerTriggers"] = args ? args.timerTriggers : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(RegistryTask.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing RegistryTask resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new RegistryTask(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of RegistryTask.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RegistryTask.__pulumiType;
    }
}
exports.RegistryTask = RegistryTask;
/** @internal */
RegistryTask.__pulumiType = 'azure:containerservice/registryTask:RegistryTask';
