// @pulumi/azure/containerservice/registryToken.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RegistryToken = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an Azure Container Registry token. Tokens are a preview feature only available in Premium SKU Container registries.
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const exampleResourceGroup = new azure.core.ResourceGroup("exampleResourceGroup", {location: "West Europe"});
 * const exampleRegistry = new azure.containerservice.Registry("exampleRegistry", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     location: exampleResourceGroup.location,
 *     sku: "Premium",
 *     adminEnabled: false,
 *     georeplicationLocations: [
 *         "East US",
 *         "West Europe",
 *     ],
 * });
 * const exampleRegistryScopeMap = new azure.containerservice.RegistryScopeMap("exampleRegistryScopeMap", {
 *     containerRegistryName: azurerm_container_registry.acr.name,
 *     resourceGroupName: azurerm_resource_group.rg.name,
 *     actions: [
 *         "repositories/repo1/content/read",
 *         "repositories/repo1/content/write",
 *     ],
 * });
 * const exampleRegistryToken = new azure.containerservice.RegistryToken("exampleRegistryToken", {
 *     containerRegistryName: azurerm_container_registry.acr.name,
 *     resourceGroupName: azurerm_resource_group.rg.name,
 *     scopeMapId: azurerm_container_registry_scope_map.map.id,
 * });
 * ```
 *
 * ## Import
 *
 * Container Registries can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:containerservice/registryToken:RegistryToken example /subscriptions/00000000-0000-0000-0000-000000000000/resourcegroups/mygroup1/providers/Microsoft.ContainerRegistry/registries/myregistry1/tokens/token1
 * ```
 */
class RegistryToken extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["containerRegistryName"] = state ? state.containerRegistryName : undefined;
            resourceInputs["enabled"] = state ? state.enabled : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["resourceGroupName"] = state ? state.resourceGroupName : undefined;
            resourceInputs["scopeMapId"] = state ? state.scopeMapId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.containerRegistryName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'containerRegistryName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.scopeMapId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'scopeMapId'");
            }
            resourceInputs["containerRegistryName"] = args ? args.containerRegistryName : undefined;
            resourceInputs["enabled"] = args ? args.enabled : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["scopeMapId"] = args ? args.scopeMapId : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(RegistryToken.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing RegistryToken resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new RegistryToken(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of RegistryToken.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RegistryToken.__pulumiType;
    }
}
exports.RegistryToken = RegistryToken;
/** @internal */
RegistryToken.__pulumiType = 'azure:containerservice/registryToken:RegistryToken';
