// @pulumi/azure/datashare/datasetDataLakeGen1.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatasetDataLakeGen1 = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Data Share Data Lake Gen1 Dataset.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 * import * as azuread from "@pulumi/azuread";
 *
 * const exampleResourceGroup = new azure.core.ResourceGroup("exampleResourceGroup", {location: "West Europe"});
 * const exampleAccount = new azure.datashare.Account("exampleAccount", {
 *     location: exampleResourceGroup.location,
 *     resourceGroupName: exampleResourceGroup.name,
 *     identity: {
 *         type: "SystemAssigned",
 *     },
 * });
 * const exampleShare = new azure.datashare.Share("exampleShare", {
 *     accountId: exampleAccount.id,
 *     kind: "CopyBased",
 * });
 * const exampleStore = new azure.datalake.Store("exampleStore", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     location: exampleResourceGroup.location,
 *     firewallState: "Disabled",
 * });
 * const exampleStoreFile = new azure.datalake.StoreFile("exampleStoreFile", {
 *     accountName: exampleStore.name,
 *     localFilePath: "./example/myfile.txt",
 *     remoteFilePath: "/example/myfile.txt",
 * });
 * const exampleServicePrincipal = azuread.getServicePrincipalOutput({
 *     displayName: exampleAccount.name,
 * });
 * const exampleAssignment = new azure.authorization.Assignment("exampleAssignment", {
 *     scope: exampleStore.id,
 *     roleDefinitionName: "Owner",
 *     principalId: exampleServicePrincipal.apply(exampleServicePrincipal => exampleServicePrincipal.objectId),
 * });
 * const exampleDatasetDataLakeGen1 = new azure.datashare.DatasetDataLakeGen1("exampleDatasetDataLakeGen1", {
 *     dataShareId: exampleShare.id,
 *     dataLakeStoreId: exampleStore.id,
 *     fileName: "myfile.txt",
 *     folderPath: "example",
 * }, {
 *     dependsOn: [exampleAssignment],
 * });
 * ```
 *
 * ## Import
 *
 * Data Share Data Lake Gen1 Datasets can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:datashare/datasetDataLakeGen1:DatasetDataLakeGen1 example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.DataShare/accounts/account1/shares/share1/dataSets/dataSet1
 * ```
 */
class DatasetDataLakeGen1 extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["dataLakeStoreId"] = state ? state.dataLakeStoreId : undefined;
            resourceInputs["dataShareId"] = state ? state.dataShareId : undefined;
            resourceInputs["displayName"] = state ? state.displayName : undefined;
            resourceInputs["fileName"] = state ? state.fileName : undefined;
            resourceInputs["folderPath"] = state ? state.folderPath : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.dataLakeStoreId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dataLakeStoreId'");
            }
            if ((!args || args.dataShareId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dataShareId'");
            }
            if ((!args || args.folderPath === undefined) && !opts.urn) {
                throw new Error("Missing required property 'folderPath'");
            }
            resourceInputs["dataLakeStoreId"] = args ? args.dataLakeStoreId : undefined;
            resourceInputs["dataShareId"] = args ? args.dataShareId : undefined;
            resourceInputs["fileName"] = args ? args.fileName : undefined;
            resourceInputs["folderPath"] = args ? args.folderPath : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["displayName"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(DatasetDataLakeGen1.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing DatasetDataLakeGen1 resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new DatasetDataLakeGen1(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DatasetDataLakeGen1.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DatasetDataLakeGen1.__pulumiType;
    }
}
exports.DatasetDataLakeGen1 = DatasetDataLakeGen1;
/** @internal */
DatasetDataLakeGen1.__pulumiType = 'azure:datashare/datasetDataLakeGen1:DatasetDataLakeGen1';
