// @pulumi/azure/datashare/datasetDataLakeGen2.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatasetDataLakeGen2 = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Data Share Data Lake Gen2 Dataset.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 * import * as azuread from "@pulumi/azuread";
 *
 * const exampleResourceGroup = new azure.core.ResourceGroup("exampleResourceGroup", {location: "West Europe"});
 * const exampleAccount = new azure.datashare.Account("exampleAccount", {
 *     location: exampleResourceGroup.location,
 *     resourceGroupName: exampleResourceGroup.name,
 *     identity: {
 *         type: "SystemAssigned",
 *     },
 * });
 * const exampleShare = new azure.datashare.Share("exampleShare", {
 *     accountId: exampleAccount.id,
 *     kind: "CopyBased",
 * });
 * const exampleStorage_accountAccount = new azure.storage.Account("exampleStorage/accountAccount", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     location: exampleResourceGroup.location,
 *     accountKind: "BlobStorage",
 *     accountTier: "Standard",
 *     accountReplicationType: "LRS",
 * });
 * const exampleDataLakeGen2Filesystem = new azure.storage.DataLakeGen2Filesystem("exampleDataLakeGen2Filesystem", {storageAccountId: exampleStorage / accountAccount.id});
 * const exampleServicePrincipal = exampleAccount.name.apply(name => azuread.getServicePrincipal({
 *     displayName: name,
 * }));
 * const exampleAssignment = new azure.authorization.Assignment("exampleAssignment", {
 *     scope: exampleStorage / accountAccount.id,
 *     roleDefinitionName: "Storage Blob Data Reader",
 *     principalId: exampleServicePrincipal.apply(exampleServicePrincipal => exampleServicePrincipal.objectId),
 * });
 * const exampleDatasetDataLakeGen2 = new azure.datashare.DatasetDataLakeGen2("exampleDatasetDataLakeGen2", {
 *     shareId: exampleShare.id,
 *     storageAccountId: exampleStorage / accountAccount.id,
 *     fileSystemName: exampleDataLakeGen2Filesystem.name,
 *     filePath: "myfile.txt",
 * }, {
 *     dependsOn: [exampleAssignment],
 * });
 * ```
 *
 * ## Import
 *
 * Data Share Data Lake Gen2 Datasets can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:datashare/datasetDataLakeGen2:DatasetDataLakeGen2 example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.DataShare/accounts/account1/shares/share1/dataSets/dataSet1
 * ```
 */
class DatasetDataLakeGen2 extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            inputs["displayName"] = state ? state.displayName : undefined;
            inputs["filePath"] = state ? state.filePath : undefined;
            inputs["fileSystemName"] = state ? state.fileSystemName : undefined;
            inputs["folderPath"] = state ? state.folderPath : undefined;
            inputs["name"] = state ? state.name : undefined;
            inputs["shareId"] = state ? state.shareId : undefined;
            inputs["storageAccountId"] = state ? state.storageAccountId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.fileSystemName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'fileSystemName'");
            }
            if ((!args || args.shareId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'shareId'");
            }
            if ((!args || args.storageAccountId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'storageAccountId'");
            }
            inputs["filePath"] = args ? args.filePath : undefined;
            inputs["fileSystemName"] = args ? args.fileSystemName : undefined;
            inputs["folderPath"] = args ? args.folderPath : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["shareId"] = args ? args.shareId : undefined;
            inputs["storageAccountId"] = args ? args.storageAccountId : undefined;
            inputs["displayName"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(DatasetDataLakeGen2.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing DatasetDataLakeGen2 resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new DatasetDataLakeGen2(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DatasetDataLakeGen2.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DatasetDataLakeGen2.__pulumiType;
    }
}
exports.DatasetDataLakeGen2 = DatasetDataLakeGen2;
/** @internal */
DatasetDataLakeGen2.__pulumiType = 'azure:datashare/datasetDataLakeGen2:DatasetDataLakeGen2';
