// @pulumi/azure/netapp/getSnapshotPolicy.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.getSnapshotPolicyOutput = exports.getSnapshotPolicy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Uses this data source to access information about an existing NetApp Snapshot Policy.
 *
 * ## NetApp Snapshot Policy Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const example = azure.netapp.getSnapshotPolicy({
 *     resourceGroupName: "acctestRG",
 *     accountName: "acctestnetappaccount",
 *     name: "example-snapshot-policy",
 * });
 * export const id = example.then(example => example.id);
 * export const name = example.then(example => example.name);
 * export const enabled = example.then(example => example.enabled);
 * export const hourlySchedule = example.then(example => example.hourlySchedules);
 * export const dailySchedule = example.then(example => example.dailySchedules);
 * export const weeklySchedule = example.then(example => example.weeklySchedules);
 * export const monthlySchedule = example.then(example => example.monthlySchedules);
 * ```
 */
function getSnapshotPolicy(args, opts) {
    if (!opts) {
        opts = {};
    }
    opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
    return pulumi.runtime.invoke("azure:netapp/getSnapshotPolicy:getSnapshotPolicy", {
        "accountName": args.accountName,
        "name": args.name,
        "resourceGroupName": args.resourceGroupName,
    }, opts);
}
exports.getSnapshotPolicy = getSnapshotPolicy;
function getSnapshotPolicyOutput(args, opts) {
    return pulumi.output(args).apply(a => getSnapshotPolicy(a, opts));
}
exports.getSnapshotPolicyOutput = getSnapshotPolicyOutput;
