// @pulumi/azure/appservice/publicCertificate.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicCertificate = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an App Service Public Certificate.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 * import * from "fs";
 *
 * const exampleResourceGroup = new azure.core.ResourceGroup("exampleResourceGroup", {location: "West Europe"});
 * const examplePlan = new azure.appservice.Plan("examplePlan", {
 *     location: exampleResourceGroup.location,
 *     resourceGroupName: exampleResourceGroup.name,
 *     sku: {
 *         tier: "Standard",
 *         size: "S1",
 *     },
 * });
 * const exampleAppService = new azure.appservice.AppService("exampleAppService", {
 *     location: exampleResourceGroup.location,
 *     resourceGroupName: exampleResourceGroup.name,
 *     appServicePlanId: examplePlan.id,
 * });
 * const examplePublicCertificate = new azure.appservice.PublicCertificate("examplePublicCertificate", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     appServiceName: exampleAppService.name,
 *     certificateName: "example-public-certificate",
 *     certificateLocation: "Unknown",
 *     blob: Buffer.from(fs.readFileSync("app_service_public_certificate.cer"), 'binary').toString('base64'),
 * });
 * ```
 *
 * ## Import
 *
 * App Service Public Certificates can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:appservice/publicCertificate:PublicCertificate example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Web/sites/site1/publicCertificates/publicCertificate1
 * ```
 */
class PublicCertificate extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["appServiceName"] = state ? state.appServiceName : undefined;
            resourceInputs["blob"] = state ? state.blob : undefined;
            resourceInputs["certificateLocation"] = state ? state.certificateLocation : undefined;
            resourceInputs["certificateName"] = state ? state.certificateName : undefined;
            resourceInputs["resourceGroupName"] = state ? state.resourceGroupName : undefined;
            resourceInputs["thumbprint"] = state ? state.thumbprint : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.appServiceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'appServiceName'");
            }
            if ((!args || args.blob === undefined) && !opts.urn) {
                throw new Error("Missing required property 'blob'");
            }
            if ((!args || args.certificateLocation === undefined) && !opts.urn) {
                throw new Error("Missing required property 'certificateLocation'");
            }
            if ((!args || args.certificateName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'certificateName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["appServiceName"] = args ? args.appServiceName : undefined;
            resourceInputs["blob"] = args ? args.blob : undefined;
            resourceInputs["certificateLocation"] = args ? args.certificateLocation : undefined;
            resourceInputs["certificateName"] = args ? args.certificateName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["thumbprint"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(PublicCertificate.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing PublicCertificate resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new PublicCertificate(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of PublicCertificate.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PublicCertificate.__pulumiType;
    }
}
exports.PublicCertificate = PublicCertificate;
/** @internal */
PublicCertificate.__pulumiType = 'azure:appservice/publicCertificate:PublicCertificate';
