// @pulumi/azuread/user.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.User = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * Manages a user within Azure Active Directory.
 *
 * ## API Permissions
 *
 * The following API permissions are required in order to use this resource.
 *
 * When authenticated with a service principal, this resource requires one of the following application roles: `User.ReadWrite.All` or `Directory.ReadWrite.All`
 *
 * When authenticated with a user principal, this resource requires one of the following directory roles: `User Administrator` or `Global Administrator`
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azuread from "@pulumi/azuread";
 *
 * const example = new azuread.User("example", {
 *     displayName: "J. Doe",
 *     mailNickname: "jdoe",
 *     password: "SecretP@sswd99!",
 *     userPrincipalName: "jdoe@hashicorp.com",
 * });
 * ```
 *
 * ## Import
 *
 * Users can be imported using their object ID, e.g.
 *
 * ```sh
 *  $ pulumi import azuread:index/user:User my_user 00000000-0000-0000-0000-000000000000
 * ```
 */
class User extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["aboutMe"] = state ? state.aboutMe : undefined;
            resourceInputs["accountEnabled"] = state ? state.accountEnabled : undefined;
            resourceInputs["ageGroup"] = state ? state.ageGroup : undefined;
            resourceInputs["businessPhones"] = state ? state.businessPhones : undefined;
            resourceInputs["city"] = state ? state.city : undefined;
            resourceInputs["companyName"] = state ? state.companyName : undefined;
            resourceInputs["consentProvidedForMinor"] = state ? state.consentProvidedForMinor : undefined;
            resourceInputs["costCenter"] = state ? state.costCenter : undefined;
            resourceInputs["country"] = state ? state.country : undefined;
            resourceInputs["creationType"] = state ? state.creationType : undefined;
            resourceInputs["department"] = state ? state.department : undefined;
            resourceInputs["disablePasswordExpiration"] = state ? state.disablePasswordExpiration : undefined;
            resourceInputs["disableStrongPassword"] = state ? state.disableStrongPassword : undefined;
            resourceInputs["displayName"] = state ? state.displayName : undefined;
            resourceInputs["division"] = state ? state.division : undefined;
            resourceInputs["employeeId"] = state ? state.employeeId : undefined;
            resourceInputs["employeeType"] = state ? state.employeeType : undefined;
            resourceInputs["externalUserState"] = state ? state.externalUserState : undefined;
            resourceInputs["faxNumber"] = state ? state.faxNumber : undefined;
            resourceInputs["forcePasswordChange"] = state ? state.forcePasswordChange : undefined;
            resourceInputs["givenName"] = state ? state.givenName : undefined;
            resourceInputs["imAddresses"] = state ? state.imAddresses : undefined;
            resourceInputs["jobTitle"] = state ? state.jobTitle : undefined;
            resourceInputs["mail"] = state ? state.mail : undefined;
            resourceInputs["mailNickname"] = state ? state.mailNickname : undefined;
            resourceInputs["managerId"] = state ? state.managerId : undefined;
            resourceInputs["mobilePhone"] = state ? state.mobilePhone : undefined;
            resourceInputs["objectId"] = state ? state.objectId : undefined;
            resourceInputs["officeLocation"] = state ? state.officeLocation : undefined;
            resourceInputs["onpremisesDistinguishedName"] = state ? state.onpremisesDistinguishedName : undefined;
            resourceInputs["onpremisesDomainName"] = state ? state.onpremisesDomainName : undefined;
            resourceInputs["onpremisesImmutableId"] = state ? state.onpremisesImmutableId : undefined;
            resourceInputs["onpremisesSamAccountName"] = state ? state.onpremisesSamAccountName : undefined;
            resourceInputs["onpremisesSecurityIdentifier"] = state ? state.onpremisesSecurityIdentifier : undefined;
            resourceInputs["onpremisesSyncEnabled"] = state ? state.onpremisesSyncEnabled : undefined;
            resourceInputs["onpremisesUserPrincipalName"] = state ? state.onpremisesUserPrincipalName : undefined;
            resourceInputs["otherMails"] = state ? state.otherMails : undefined;
            resourceInputs["password"] = state ? state.password : undefined;
            resourceInputs["postalCode"] = state ? state.postalCode : undefined;
            resourceInputs["preferredLanguage"] = state ? state.preferredLanguage : undefined;
            resourceInputs["proxyAddresses"] = state ? state.proxyAddresses : undefined;
            resourceInputs["showInAddressList"] = state ? state.showInAddressList : undefined;
            resourceInputs["state"] = state ? state.state : undefined;
            resourceInputs["streetAddress"] = state ? state.streetAddress : undefined;
            resourceInputs["surname"] = state ? state.surname : undefined;
            resourceInputs["usageLocation"] = state ? state.usageLocation : undefined;
            resourceInputs["userPrincipalName"] = state ? state.userPrincipalName : undefined;
            resourceInputs["userType"] = state ? state.userType : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.displayName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'displayName'");
            }
            if ((!args || args.userPrincipalName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'userPrincipalName'");
            }
            resourceInputs["accountEnabled"] = args ? args.accountEnabled : undefined;
            resourceInputs["ageGroup"] = args ? args.ageGroup : undefined;
            resourceInputs["businessPhones"] = args ? args.businessPhones : undefined;
            resourceInputs["city"] = args ? args.city : undefined;
            resourceInputs["companyName"] = args ? args.companyName : undefined;
            resourceInputs["consentProvidedForMinor"] = args ? args.consentProvidedForMinor : undefined;
            resourceInputs["costCenter"] = args ? args.costCenter : undefined;
            resourceInputs["country"] = args ? args.country : undefined;
            resourceInputs["department"] = args ? args.department : undefined;
            resourceInputs["disablePasswordExpiration"] = args ? args.disablePasswordExpiration : undefined;
            resourceInputs["disableStrongPassword"] = args ? args.disableStrongPassword : undefined;
            resourceInputs["displayName"] = args ? args.displayName : undefined;
            resourceInputs["division"] = args ? args.division : undefined;
            resourceInputs["employeeId"] = args ? args.employeeId : undefined;
            resourceInputs["employeeType"] = args ? args.employeeType : undefined;
            resourceInputs["faxNumber"] = args ? args.faxNumber : undefined;
            resourceInputs["forcePasswordChange"] = args ? args.forcePasswordChange : undefined;
            resourceInputs["givenName"] = args ? args.givenName : undefined;
            resourceInputs["jobTitle"] = args ? args.jobTitle : undefined;
            resourceInputs["mail"] = args ? args.mail : undefined;
            resourceInputs["mailNickname"] = args ? args.mailNickname : undefined;
            resourceInputs["managerId"] = args ? args.managerId : undefined;
            resourceInputs["mobilePhone"] = args ? args.mobilePhone : undefined;
            resourceInputs["officeLocation"] = args ? args.officeLocation : undefined;
            resourceInputs["onpremisesImmutableId"] = args ? args.onpremisesImmutableId : undefined;
            resourceInputs["otherMails"] = args ? args.otherMails : undefined;
            resourceInputs["password"] = args ? args.password : undefined;
            resourceInputs["postalCode"] = args ? args.postalCode : undefined;
            resourceInputs["preferredLanguage"] = args ? args.preferredLanguage : undefined;
            resourceInputs["showInAddressList"] = args ? args.showInAddressList : undefined;
            resourceInputs["state"] = args ? args.state : undefined;
            resourceInputs["streetAddress"] = args ? args.streetAddress : undefined;
            resourceInputs["surname"] = args ? args.surname : undefined;
            resourceInputs["usageLocation"] = args ? args.usageLocation : undefined;
            resourceInputs["userPrincipalName"] = args ? args.userPrincipalName : undefined;
            resourceInputs["aboutMe"] = undefined /*out*/;
            resourceInputs["creationType"] = undefined /*out*/;
            resourceInputs["externalUserState"] = undefined /*out*/;
            resourceInputs["imAddresses"] = undefined /*out*/;
            resourceInputs["objectId"] = undefined /*out*/;
            resourceInputs["onpremisesDistinguishedName"] = undefined /*out*/;
            resourceInputs["onpremisesDomainName"] = undefined /*out*/;
            resourceInputs["onpremisesSamAccountName"] = undefined /*out*/;
            resourceInputs["onpremisesSecurityIdentifier"] = undefined /*out*/;
            resourceInputs["onpremisesSyncEnabled"] = undefined /*out*/;
            resourceInputs["onpremisesUserPrincipalName"] = undefined /*out*/;
            resourceInputs["proxyAddresses"] = undefined /*out*/;
            resourceInputs["userType"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(User.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing User resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new User(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of User.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === User.__pulumiType;
    }
}
exports.User = User;
/** @internal */
User.__pulumiType = 'azuread:index/user:User';
