// @pulumi/azure/digitaltwins/endpointEventGrid.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.EndpointEventGrid = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Digital Twins Event Grid Endpoint.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const exampleResourceGroup = new azure.core.ResourceGroup("exampleResourceGroup", {location: "West Europe"});
 * const exampleInstance = new azure.digitaltwins.Instance("exampleInstance", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     location: exampleResourceGroup.location,
 * });
 * const exampleTopic = new azure.eventgrid.Topic("exampleTopic", {
 *     location: exampleResourceGroup.location,
 *     resourceGroupName: exampleResourceGroup.name,
 * });
 * const exampleEndpointEventGrid = new azure.digitaltwins.EndpointEventGrid("exampleEndpointEventGrid", {
 *     digitalTwinsId: exampleInstance.id,
 *     eventgridTopicEndpoint: exampleTopic.endpoint,
 *     eventgridTopicPrimaryAccessKey: exampleTopic.primaryAccessKey,
 *     eventgridTopicSecondaryAccessKey: exampleTopic.secondaryAccessKey,
 * });
 * ```
 *
 * ## Import
 *
 * Digital Twins Eventgrid Endpoints can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:digitaltwins/endpointEventGrid:EndpointEventGrid example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.DigitalTwins/digitalTwinsInstances/dt1/endpoints/ep1
 * ```
 */
class EndpointEventGrid extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["deadLetterStorageSecret"] = state ? state.deadLetterStorageSecret : undefined;
            resourceInputs["digitalTwinsId"] = state ? state.digitalTwinsId : undefined;
            resourceInputs["eventgridTopicEndpoint"] = state ? state.eventgridTopicEndpoint : undefined;
            resourceInputs["eventgridTopicPrimaryAccessKey"] = state ? state.eventgridTopicPrimaryAccessKey : undefined;
            resourceInputs["eventgridTopicSecondaryAccessKey"] = state ? state.eventgridTopicSecondaryAccessKey : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.digitalTwinsId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'digitalTwinsId'");
            }
            if ((!args || args.eventgridTopicEndpoint === undefined) && !opts.urn) {
                throw new Error("Missing required property 'eventgridTopicEndpoint'");
            }
            if ((!args || args.eventgridTopicPrimaryAccessKey === undefined) && !opts.urn) {
                throw new Error("Missing required property 'eventgridTopicPrimaryAccessKey'");
            }
            if ((!args || args.eventgridTopicSecondaryAccessKey === undefined) && !opts.urn) {
                throw new Error("Missing required property 'eventgridTopicSecondaryAccessKey'");
            }
            resourceInputs["deadLetterStorageSecret"] = args ? args.deadLetterStorageSecret : undefined;
            resourceInputs["digitalTwinsId"] = args ? args.digitalTwinsId : undefined;
            resourceInputs["eventgridTopicEndpoint"] = args ? args.eventgridTopicEndpoint : undefined;
            resourceInputs["eventgridTopicPrimaryAccessKey"] = args ? args.eventgridTopicPrimaryAccessKey : undefined;
            resourceInputs["eventgridTopicSecondaryAccessKey"] = args ? args.eventgridTopicSecondaryAccessKey : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(EndpointEventGrid.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing EndpointEventGrid resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new EndpointEventGrid(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of EndpointEventGrid.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EndpointEventGrid.__pulumiType;
    }
}
exports.EndpointEventGrid = EndpointEventGrid;
/** @internal */
EndpointEventGrid.__pulumiType = 'azure:digitaltwins/endpointEventGrid:EndpointEventGrid';
