// @pulumi/azure/bot/serviceAzureBot.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceAzureBot = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an Azure Bot Service.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const exampleResourceGroup = new azure.core.ResourceGroup("exampleResourceGroup", {location: "West Europe"});
 * const exampleInsights = new azure.appinsights.Insights("exampleInsights", {
 *     location: exampleResourceGroup.location,
 *     resourceGroupName: exampleResourceGroup.name,
 *     applicationType: "web",
 * });
 * const exampleApiKey = new azure.appinsights.ApiKey("exampleApiKey", {
 *     applicationInsightsId: exampleInsights.id,
 *     readPermissions: [
 *         "aggregate",
 *         "api",
 *         "draft",
 *         "extendqueries",
 *         "search",
 *     ],
 * });
 * const exampleServiceAzureBot = new azure.bot.ServiceAzureBot("exampleServiceAzureBot", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     location: "global",
 *     microsoftAppId: data.azurerm_client_config.current.client_id,
 *     sku: "F0",
 *     endpoint: "https://example.com",
 *     developerAppInsightsApiKey: exampleApiKey.apiKey,
 *     developerAppInsightsApplicationId: exampleInsights.appId,
 *     tags: {
 *         environment: "test",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Azure Bot Services can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:bot/serviceAzureBot:ServiceAzureBot example /subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/resGroup1/providers/Microsoft.BotService/botServices/botService1
 * ```
 */
class ServiceAzureBot extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            inputs["developerAppInsightsApiKey"] = state ? state.developerAppInsightsApiKey : undefined;
            inputs["developerAppInsightsApplicationId"] = state ? state.developerAppInsightsApplicationId : undefined;
            inputs["developerAppInsightsKey"] = state ? state.developerAppInsightsKey : undefined;
            inputs["displayName"] = state ? state.displayName : undefined;
            inputs["endpoint"] = state ? state.endpoint : undefined;
            inputs["location"] = state ? state.location : undefined;
            inputs["luisAppIds"] = state ? state.luisAppIds : undefined;
            inputs["luisKey"] = state ? state.luisKey : undefined;
            inputs["microsoftAppId"] = state ? state.microsoftAppId : undefined;
            inputs["name"] = state ? state.name : undefined;
            inputs["resourceGroupName"] = state ? state.resourceGroupName : undefined;
            inputs["sku"] = state ? state.sku : undefined;
            inputs["tags"] = state ? state.tags : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.microsoftAppId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'microsoftAppId'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.sku === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sku'");
            }
            inputs["developerAppInsightsApiKey"] = args ? args.developerAppInsightsApiKey : undefined;
            inputs["developerAppInsightsApplicationId"] = args ? args.developerAppInsightsApplicationId : undefined;
            inputs["developerAppInsightsKey"] = args ? args.developerAppInsightsKey : undefined;
            inputs["displayName"] = args ? args.displayName : undefined;
            inputs["endpoint"] = args ? args.endpoint : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["luisAppIds"] = args ? args.luisAppIds : undefined;
            inputs["luisKey"] = args ? args.luisKey : undefined;
            inputs["microsoftAppId"] = args ? args.microsoftAppId : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["sku"] = args ? args.sku : undefined;
            inputs["tags"] = args ? args.tags : undefined;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(ServiceAzureBot.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ServiceAzureBot resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ServiceAzureBot(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ServiceAzureBot.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ServiceAzureBot.__pulumiType;
    }
}
exports.ServiceAzureBot = ServiceAzureBot;
/** @internal */
ServiceAzureBot.__pulumiType = 'azure:bot/serviceAzureBot:ServiceAzureBot';
