// @pulumi/azure/storage/account.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Account = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an Azure Storage Account.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const exampleResourceGroup = new azure.core.ResourceGroup("exampleResourceGroup", {location: "West Europe"});
 * const exampleAccount = new azure.storage.Account("exampleAccount", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     location: exampleResourceGroup.location,
 *     accountTier: "Standard",
 *     accountReplicationType: "GRS",
 *     tags: {
 *         environment: "staging",
 *     },
 * });
 * ```
 * ### With Network Rules
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const exampleResourceGroup = new azure.core.ResourceGroup("exampleResourceGroup", {location: "West Europe"});
 * const exampleVirtualNetwork = new azure.network.VirtualNetwork("exampleVirtualNetwork", {
 *     addressSpaces: ["10.0.0.0/16"],
 *     location: exampleResourceGroup.location,
 *     resourceGroupName: exampleResourceGroup.name,
 * });
 * const exampleSubnet = new azure.network.Subnet("exampleSubnet", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     virtualNetworkName: exampleVirtualNetwork.name,
 *     addressPrefixes: ["10.0.2.0/24"],
 *     serviceEndpoints: [
 *         "Microsoft.Sql",
 *         "Microsoft.Storage",
 *     ],
 * });
 * const exampleAccount = new azure.storage.Account("exampleAccount", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     location: exampleResourceGroup.location,
 *     accountTier: "Standard",
 *     accountReplicationType: "LRS",
 *     networkRules: {
 *         defaultAction: "Deny",
 *         ipRules: ["100.0.0.1"],
 *         virtualNetworkSubnetIds: [exampleSubnet.id],
 *     },
 *     tags: {
 *         environment: "staging",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Storage Accounts can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:storage/account:Account storageAcc1 /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/myresourcegroup/providers/Microsoft.Storage/storageAccounts/myaccount
 * ```
 */
class Account extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accessTier"] = state ? state.accessTier : undefined;
            resourceInputs["accountKind"] = state ? state.accountKind : undefined;
            resourceInputs["accountReplicationType"] = state ? state.accountReplicationType : undefined;
            resourceInputs["accountTier"] = state ? state.accountTier : undefined;
            resourceInputs["allowBlobPublicAccess"] = state ? state.allowBlobPublicAccess : undefined;
            resourceInputs["azureFilesAuthentication"] = state ? state.azureFilesAuthentication : undefined;
            resourceInputs["blobProperties"] = state ? state.blobProperties : undefined;
            resourceInputs["customDomain"] = state ? state.customDomain : undefined;
            resourceInputs["customerManagedKey"] = state ? state.customerManagedKey : undefined;
            resourceInputs["enableHttpsTrafficOnly"] = state ? state.enableHttpsTrafficOnly : undefined;
            resourceInputs["identity"] = state ? state.identity : undefined;
            resourceInputs["infrastructureEncryptionEnabled"] = state ? state.infrastructureEncryptionEnabled : undefined;
            resourceInputs["isHnsEnabled"] = state ? state.isHnsEnabled : undefined;
            resourceInputs["largeFileShareEnabled"] = state ? state.largeFileShareEnabled : undefined;
            resourceInputs["location"] = state ? state.location : undefined;
            resourceInputs["minTlsVersion"] = state ? state.minTlsVersion : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["networkRules"] = state ? state.networkRules : undefined;
            resourceInputs["nfsv3Enabled"] = state ? state.nfsv3Enabled : undefined;
            resourceInputs["primaryAccessKey"] = state ? state.primaryAccessKey : undefined;
            resourceInputs["primaryBlobConnectionString"] = state ? state.primaryBlobConnectionString : undefined;
            resourceInputs["primaryBlobEndpoint"] = state ? state.primaryBlobEndpoint : undefined;
            resourceInputs["primaryBlobHost"] = state ? state.primaryBlobHost : undefined;
            resourceInputs["primaryConnectionString"] = state ? state.primaryConnectionString : undefined;
            resourceInputs["primaryDfsEndpoint"] = state ? state.primaryDfsEndpoint : undefined;
            resourceInputs["primaryDfsHost"] = state ? state.primaryDfsHost : undefined;
            resourceInputs["primaryFileEndpoint"] = state ? state.primaryFileEndpoint : undefined;
            resourceInputs["primaryFileHost"] = state ? state.primaryFileHost : undefined;
            resourceInputs["primaryLocation"] = state ? state.primaryLocation : undefined;
            resourceInputs["primaryQueueEndpoint"] = state ? state.primaryQueueEndpoint : undefined;
            resourceInputs["primaryQueueHost"] = state ? state.primaryQueueHost : undefined;
            resourceInputs["primaryTableEndpoint"] = state ? state.primaryTableEndpoint : undefined;
            resourceInputs["primaryTableHost"] = state ? state.primaryTableHost : undefined;
            resourceInputs["primaryWebEndpoint"] = state ? state.primaryWebEndpoint : undefined;
            resourceInputs["primaryWebHost"] = state ? state.primaryWebHost : undefined;
            resourceInputs["queueEncryptionKeyType"] = state ? state.queueEncryptionKeyType : undefined;
            resourceInputs["queueProperties"] = state ? state.queueProperties : undefined;
            resourceInputs["resourceGroupName"] = state ? state.resourceGroupName : undefined;
            resourceInputs["routing"] = state ? state.routing : undefined;
            resourceInputs["secondaryAccessKey"] = state ? state.secondaryAccessKey : undefined;
            resourceInputs["secondaryBlobConnectionString"] = state ? state.secondaryBlobConnectionString : undefined;
            resourceInputs["secondaryBlobEndpoint"] = state ? state.secondaryBlobEndpoint : undefined;
            resourceInputs["secondaryBlobHost"] = state ? state.secondaryBlobHost : undefined;
            resourceInputs["secondaryConnectionString"] = state ? state.secondaryConnectionString : undefined;
            resourceInputs["secondaryDfsEndpoint"] = state ? state.secondaryDfsEndpoint : undefined;
            resourceInputs["secondaryDfsHost"] = state ? state.secondaryDfsHost : undefined;
            resourceInputs["secondaryFileEndpoint"] = state ? state.secondaryFileEndpoint : undefined;
            resourceInputs["secondaryFileHost"] = state ? state.secondaryFileHost : undefined;
            resourceInputs["secondaryLocation"] = state ? state.secondaryLocation : undefined;
            resourceInputs["secondaryQueueEndpoint"] = state ? state.secondaryQueueEndpoint : undefined;
            resourceInputs["secondaryQueueHost"] = state ? state.secondaryQueueHost : undefined;
            resourceInputs["secondaryTableEndpoint"] = state ? state.secondaryTableEndpoint : undefined;
            resourceInputs["secondaryTableHost"] = state ? state.secondaryTableHost : undefined;
            resourceInputs["secondaryWebEndpoint"] = state ? state.secondaryWebEndpoint : undefined;
            resourceInputs["secondaryWebHost"] = state ? state.secondaryWebHost : undefined;
            resourceInputs["shareProperties"] = state ? state.shareProperties : undefined;
            resourceInputs["sharedAccessKeyEnabled"] = state ? state.sharedAccessKeyEnabled : undefined;
            resourceInputs["staticWebsite"] = state ? state.staticWebsite : undefined;
            resourceInputs["tableEncryptionKeyType"] = state ? state.tableEncryptionKeyType : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.accountReplicationType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountReplicationType'");
            }
            if ((!args || args.accountTier === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountTier'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["accessTier"] = args ? args.accessTier : undefined;
            resourceInputs["accountKind"] = args ? args.accountKind : undefined;
            resourceInputs["accountReplicationType"] = args ? args.accountReplicationType : undefined;
            resourceInputs["accountTier"] = args ? args.accountTier : undefined;
            resourceInputs["allowBlobPublicAccess"] = args ? args.allowBlobPublicAccess : undefined;
            resourceInputs["azureFilesAuthentication"] = args ? args.azureFilesAuthentication : undefined;
            resourceInputs["blobProperties"] = args ? args.blobProperties : undefined;
            resourceInputs["customDomain"] = args ? args.customDomain : undefined;
            resourceInputs["customerManagedKey"] = args ? args.customerManagedKey : undefined;
            resourceInputs["enableHttpsTrafficOnly"] = args ? args.enableHttpsTrafficOnly : undefined;
            resourceInputs["identity"] = args ? args.identity : undefined;
            resourceInputs["infrastructureEncryptionEnabled"] = args ? args.infrastructureEncryptionEnabled : undefined;
            resourceInputs["isHnsEnabled"] = args ? args.isHnsEnabled : undefined;
            resourceInputs["largeFileShareEnabled"] = args ? args.largeFileShareEnabled : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["minTlsVersion"] = args ? args.minTlsVersion : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["networkRules"] = args ? args.networkRules : undefined;
            resourceInputs["nfsv3Enabled"] = args ? args.nfsv3Enabled : undefined;
            resourceInputs["queueEncryptionKeyType"] = args ? args.queueEncryptionKeyType : undefined;
            resourceInputs["queueProperties"] = args ? args.queueProperties : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["routing"] = args ? args.routing : undefined;
            resourceInputs["shareProperties"] = args ? args.shareProperties : undefined;
            resourceInputs["sharedAccessKeyEnabled"] = args ? args.sharedAccessKeyEnabled : undefined;
            resourceInputs["staticWebsite"] = args ? args.staticWebsite : undefined;
            resourceInputs["tableEncryptionKeyType"] = args ? args.tableEncryptionKeyType : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["primaryAccessKey"] = undefined /*out*/;
            resourceInputs["primaryBlobConnectionString"] = undefined /*out*/;
            resourceInputs["primaryBlobEndpoint"] = undefined /*out*/;
            resourceInputs["primaryBlobHost"] = undefined /*out*/;
            resourceInputs["primaryConnectionString"] = undefined /*out*/;
            resourceInputs["primaryDfsEndpoint"] = undefined /*out*/;
            resourceInputs["primaryDfsHost"] = undefined /*out*/;
            resourceInputs["primaryFileEndpoint"] = undefined /*out*/;
            resourceInputs["primaryFileHost"] = undefined /*out*/;
            resourceInputs["primaryLocation"] = undefined /*out*/;
            resourceInputs["primaryQueueEndpoint"] = undefined /*out*/;
            resourceInputs["primaryQueueHost"] = undefined /*out*/;
            resourceInputs["primaryTableEndpoint"] = undefined /*out*/;
            resourceInputs["primaryTableHost"] = undefined /*out*/;
            resourceInputs["primaryWebEndpoint"] = undefined /*out*/;
            resourceInputs["primaryWebHost"] = undefined /*out*/;
            resourceInputs["secondaryAccessKey"] = undefined /*out*/;
            resourceInputs["secondaryBlobConnectionString"] = undefined /*out*/;
            resourceInputs["secondaryBlobEndpoint"] = undefined /*out*/;
            resourceInputs["secondaryBlobHost"] = undefined /*out*/;
            resourceInputs["secondaryConnectionString"] = undefined /*out*/;
            resourceInputs["secondaryDfsEndpoint"] = undefined /*out*/;
            resourceInputs["secondaryDfsHost"] = undefined /*out*/;
            resourceInputs["secondaryFileEndpoint"] = undefined /*out*/;
            resourceInputs["secondaryFileHost"] = undefined /*out*/;
            resourceInputs["secondaryLocation"] = undefined /*out*/;
            resourceInputs["secondaryQueueEndpoint"] = undefined /*out*/;
            resourceInputs["secondaryQueueHost"] = undefined /*out*/;
            resourceInputs["secondaryTableEndpoint"] = undefined /*out*/;
            resourceInputs["secondaryTableHost"] = undefined /*out*/;
            resourceInputs["secondaryWebEndpoint"] = undefined /*out*/;
            resourceInputs["secondaryWebHost"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Account.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Account resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Account(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Account.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Account.__pulumiType;
    }
}
exports.Account = Account;
/** @internal */
Account.__pulumiType = 'azure:storage/account:Account';
