// @pulumi/azure/appservice/environmentV3.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnvironmentV3 = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * ## Import
 *
 * A 3rd Generation (v3) App Service Environment can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:appservice/environmentV3:EnvironmentV3 myAppServiceEnv /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/myResourceGroup/providers/Microsoft.Web/hostingEnvironments/myAppServiceEnv
 * ```
 */
class EnvironmentV3 extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            inputs["allowNewPrivateEndpointConnections"] = state ? state.allowNewPrivateEndpointConnections : undefined;
            inputs["clusterSettings"] = state ? state.clusterSettings : undefined;
            inputs["dedicatedHostCount"] = state ? state.dedicatedHostCount : undefined;
            inputs["dnsSuffix"] = state ? state.dnsSuffix : undefined;
            inputs["externalInboundIpAddresses"] = state ? state.externalInboundIpAddresses : undefined;
            inputs["inboundNetworkDependencies"] = state ? state.inboundNetworkDependencies : undefined;
            inputs["internalInboundIpAddresses"] = state ? state.internalInboundIpAddresses : undefined;
            inputs["internalLoadBalancingMode"] = state ? state.internalLoadBalancingMode : undefined;
            inputs["ipSslAddressCount"] = state ? state.ipSslAddressCount : undefined;
            inputs["linuxOutboundIpAddresses"] = state ? state.linuxOutboundIpAddresses : undefined;
            inputs["location"] = state ? state.location : undefined;
            inputs["name"] = state ? state.name : undefined;
            inputs["pricingTier"] = state ? state.pricingTier : undefined;
            inputs["resourceGroupName"] = state ? state.resourceGroupName : undefined;
            inputs["subnetId"] = state ? state.subnetId : undefined;
            inputs["tags"] = state ? state.tags : undefined;
            inputs["windowsOutboundIpAddresses"] = state ? state.windowsOutboundIpAddresses : undefined;
            inputs["zoneRedundant"] = state ? state.zoneRedundant : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.subnetId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'subnetId'");
            }
            inputs["allowNewPrivateEndpointConnections"] = args ? args.allowNewPrivateEndpointConnections : undefined;
            inputs["clusterSettings"] = args ? args.clusterSettings : undefined;
            inputs["dedicatedHostCount"] = args ? args.dedicatedHostCount : undefined;
            inputs["internalLoadBalancingMode"] = args ? args.internalLoadBalancingMode : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["subnetId"] = args ? args.subnetId : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["zoneRedundant"] = args ? args.zoneRedundant : undefined;
            inputs["dnsSuffix"] = undefined /*out*/;
            inputs["externalInboundIpAddresses"] = undefined /*out*/;
            inputs["inboundNetworkDependencies"] = undefined /*out*/;
            inputs["internalInboundIpAddresses"] = undefined /*out*/;
            inputs["ipSslAddressCount"] = undefined /*out*/;
            inputs["linuxOutboundIpAddresses"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["pricingTier"] = undefined /*out*/;
            inputs["windowsOutboundIpAddresses"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(EnvironmentV3.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing EnvironmentV3 resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new EnvironmentV3(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of EnvironmentV3.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EnvironmentV3.__pulumiType;
    }
}
exports.EnvironmentV3 = EnvironmentV3;
/** @internal */
EnvironmentV3.__pulumiType = 'azure:appservice/environmentV3:EnvironmentV3';
