// @pulumi/azure/bot/channelDirectLineSpeech.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChannelDirectLineSpeech = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Direct Line Speech integration for a Bot Channel
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const current = azure.core.getClientConfig({});
 * const exampleResourceGroup = new azure.core.ResourceGroup("exampleResourceGroup", {location: "West Europe"});
 * const exampleAccount = new azure.cognitive.Account("exampleAccount", {
 *     location: exampleResourceGroup.location,
 *     resourceGroupName: exampleResourceGroup.name,
 *     kind: "SpeechServices",
 *     skuName: "S0",
 * });
 * const exampleChannelsRegistration = new azure.bot.ChannelsRegistration("exampleChannelsRegistration", {
 *     location: "global",
 *     resourceGroupName: exampleResourceGroup.name,
 *     sku: "F0",
 *     microsoftAppId: current.then(current => current.clientId),
 * });
 * const exampleChannelDirectLineSpeech = new azure.bot.ChannelDirectLineSpeech("exampleChannelDirectLineSpeech", {
 *     botName: exampleChannelsRegistration.name,
 *     location: exampleChannelsRegistration.location,
 *     resourceGroupName: exampleResourceGroup.name,
 *     cognitiveServiceLocation: exampleAccount.location,
 *     cognitiveServiceAccessKey: exampleAccount.primaryAccessKey,
 * });
 * ```
 *
 * ## Import
 *
 * Direct Line Speech Channels can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:bot/channelDirectLineSpeech:ChannelDirectLineSpeech example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.BotService/botServices/botService1/channels/DirectLineSpeechChannel
 * ```
 */
class ChannelDirectLineSpeech extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["botName"] = state ? state.botName : undefined;
            resourceInputs["cognitiveServiceAccessKey"] = state ? state.cognitiveServiceAccessKey : undefined;
            resourceInputs["cognitiveServiceLocation"] = state ? state.cognitiveServiceLocation : undefined;
            resourceInputs["customSpeechModelId"] = state ? state.customSpeechModelId : undefined;
            resourceInputs["customVoiceDeploymentId"] = state ? state.customVoiceDeploymentId : undefined;
            resourceInputs["location"] = state ? state.location : undefined;
            resourceInputs["resourceGroupName"] = state ? state.resourceGroupName : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.botName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'botName'");
            }
            if ((!args || args.cognitiveServiceAccessKey === undefined) && !opts.urn) {
                throw new Error("Missing required property 'cognitiveServiceAccessKey'");
            }
            if ((!args || args.cognitiveServiceLocation === undefined) && !opts.urn) {
                throw new Error("Missing required property 'cognitiveServiceLocation'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["botName"] = args ? args.botName : undefined;
            resourceInputs["cognitiveServiceAccessKey"] = args ? args.cognitiveServiceAccessKey : undefined;
            resourceInputs["cognitiveServiceLocation"] = args ? args.cognitiveServiceLocation : undefined;
            resourceInputs["customSpeechModelId"] = args ? args.customSpeechModelId : undefined;
            resourceInputs["customVoiceDeploymentId"] = args ? args.customVoiceDeploymentId : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ChannelDirectLineSpeech.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing ChannelDirectLineSpeech resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ChannelDirectLineSpeech(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ChannelDirectLineSpeech.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ChannelDirectLineSpeech.__pulumiType;
    }
}
exports.ChannelDirectLineSpeech = ChannelDirectLineSpeech;
/** @internal */
ChannelDirectLineSpeech.__pulumiType = 'azure:bot/channelDirectLineSpeech:ChannelDirectLineSpeech';
