// @pulumi/azure/maps/creator.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Creator = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an Azure Maps Creator.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const exampleResourceGroup = new azure.core.ResourceGroup("exampleResourceGroup", {location: "West Europe"});
 * const exampleAccount = new azure.maps.Account("exampleAccount", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     skuName: "G2",
 *     tags: {
 *         environment: "Test",
 *     },
 * });
 * const exampleCreator = new azure.maps.Creator("exampleCreator", {
 *     mapsAccountId: exampleAccount.id,
 *     location: exampleResourceGroup.location,
 *     storageUnits: 1,
 *     tags: {
 *         environment: "Test",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * An Azure Maps Creators can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:maps/creator:Creator example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resourceGroup1/providers/Microsoft.Maps/accounts/account1/creators/creator1
 * ```
 */
class Creator extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["location"] = state ? state.location : undefined;
            resourceInputs["mapsAccountId"] = state ? state.mapsAccountId : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["storageUnits"] = state ? state.storageUnits : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.mapsAccountId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'mapsAccountId'");
            }
            if ((!args || args.storageUnits === undefined) && !opts.urn) {
                throw new Error("Missing required property 'storageUnits'");
            }
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["mapsAccountId"] = args ? args.mapsAccountId : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["storageUnits"] = args ? args.storageUnits : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Creator.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Creator resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Creator(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Creator.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Creator.__pulumiType;
    }
}
exports.Creator = Creator;
/** @internal */
Creator.__pulumiType = 'azure:maps/creator:Creator';
