// @pulumi/azure/apimanagement/identityProviderAadb2c.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.IdentityProviderAadb2c = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an API Management Azure AD B2C Identity Provider.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 * import * as azuread from "@pulumi/azuread";
 *
 * const exampleResourceGroup = new azure.core.ResourceGroup("exampleResourceGroup", {location: "West Europe"});
 * const exampleService = new azure.apimanagement.Service("exampleService", {
 *     location: exampleResourceGroup.location,
 *     resourceGroupName: exampleResourceGroup.name,
 *     publisherName: "My Company",
 *     publisherEmail: "company@terraform.io",
 *     skuName: "Developer_1",
 * });
 * const exampleApplication = new azuread.Application("exampleApplication", {
 *     oauth2AllowImplicitFlow: true,
 *     replyUrls: [`https://${azurerm_api_management.test.name}.developer.azure-api.net/signin`],
 * });
 * const exampleApplicationPassword = new azuread.ApplicationPassword("exampleApplicationPassword", {
 *     applicationObjectId: azuread_application.test.object_id,
 *     endDateRelative: "36h",
 *     value: `P@55w0rD!%[7]s`,
 * });
 * const exampleIdentityProviderAadb2c = new azure.apimanagement.IdentityProviderAadb2c("exampleIdentityProviderAadb2c", {
 *     apiManagementName: exampleService.name,
 *     clientId: exampleApplication.applicationId,
 *     clientSecret: `P@55w0rD!%[7]s`,
 *     allowedTenant: "myb2ctenant.onmicrosoft.com",
 *     signinTenant: "myb2ctenant.onmicrosoft.com",
 *     authority: "myb2ctenant.b2clogin.com",
 *     signinPolicy: "B2C_1_Login",
 *     signupPolicy: "B2C_1_Signup",
 * }, {
 *     dependsOn: [exampleApplicationPassword],
 * });
 * ```
 *
 * ## Import
 *
 * API Management Azure AD B2C Identity Providers can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:apimanagement/identityProviderAadb2c:IdentityProviderAadb2c example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/Microsoft.ApiManagement/service/service1/identityProviders/AadB2C
 * ```
 */
class IdentityProviderAadb2c extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            inputs["allowedTenant"] = state ? state.allowedTenant : undefined;
            inputs["apiManagementName"] = state ? state.apiManagementName : undefined;
            inputs["authority"] = state ? state.authority : undefined;
            inputs["clientId"] = state ? state.clientId : undefined;
            inputs["clientSecret"] = state ? state.clientSecret : undefined;
            inputs["passwordResetPolicy"] = state ? state.passwordResetPolicy : undefined;
            inputs["profileEditingPolicy"] = state ? state.profileEditingPolicy : undefined;
            inputs["resourceGroupName"] = state ? state.resourceGroupName : undefined;
            inputs["signinPolicy"] = state ? state.signinPolicy : undefined;
            inputs["signinTenant"] = state ? state.signinTenant : undefined;
            inputs["signupPolicy"] = state ? state.signupPolicy : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.allowedTenant === undefined) && !opts.urn) {
                throw new Error("Missing required property 'allowedTenant'");
            }
            if ((!args || args.apiManagementName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'apiManagementName'");
            }
            if ((!args || args.authority === undefined) && !opts.urn) {
                throw new Error("Missing required property 'authority'");
            }
            if ((!args || args.clientId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'clientId'");
            }
            if ((!args || args.clientSecret === undefined) && !opts.urn) {
                throw new Error("Missing required property 'clientSecret'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.signinPolicy === undefined) && !opts.urn) {
                throw new Error("Missing required property 'signinPolicy'");
            }
            if ((!args || args.signinTenant === undefined) && !opts.urn) {
                throw new Error("Missing required property 'signinTenant'");
            }
            if ((!args || args.signupPolicy === undefined) && !opts.urn) {
                throw new Error("Missing required property 'signupPolicy'");
            }
            inputs["allowedTenant"] = args ? args.allowedTenant : undefined;
            inputs["apiManagementName"] = args ? args.apiManagementName : undefined;
            inputs["authority"] = args ? args.authority : undefined;
            inputs["clientId"] = args ? args.clientId : undefined;
            inputs["clientSecret"] = args ? args.clientSecret : undefined;
            inputs["passwordResetPolicy"] = args ? args.passwordResetPolicy : undefined;
            inputs["profileEditingPolicy"] = args ? args.profileEditingPolicy : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["signinPolicy"] = args ? args.signinPolicy : undefined;
            inputs["signinTenant"] = args ? args.signinTenant : undefined;
            inputs["signupPolicy"] = args ? args.signupPolicy : undefined;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(IdentityProviderAadb2c.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing IdentityProviderAadb2c resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new IdentityProviderAadb2c(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of IdentityProviderAadb2c.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === IdentityProviderAadb2c.__pulumiType;
    }
}
exports.IdentityProviderAadb2c = IdentityProviderAadb2c;
/** @internal */
IdentityProviderAadb2c.__pulumiType = 'azure:apimanagement/identityProviderAadb2c:IdentityProviderAadb2c';
