// @pulumi/azure/synapse/sqlPoolWorkloadClassifier.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlPoolWorkloadClassifier = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Synapse SQL Pool Workload Classifier.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const exampleResourceGroup = new azure.core.ResourceGroup("exampleResourceGroup", {location: "West Europe"});
 * const exampleAccount = new azure.storage.Account("exampleAccount", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     location: exampleResourceGroup.location,
 *     accountKind: "BlobStorage",
 *     accountTier: "Standard",
 *     accountReplicationType: "LRS",
 * });
 * const exampleDataLakeGen2Filesystem = new azure.storage.DataLakeGen2Filesystem("exampleDataLakeGen2Filesystem", {storageAccountId: exampleAccount.id});
 * const exampleWorkspace = new azure.synapse.Workspace("exampleWorkspace", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     location: exampleResourceGroup.location,
 *     storageDataLakeGen2FilesystemId: exampleDataLakeGen2Filesystem.id,
 *     sqlAdministratorLogin: "sqladminuser",
 *     sqlAdministratorLoginPassword: "H@Sh1CoR3!",
 * });
 * const exampleSqlPool = new azure.synapse.SqlPool("exampleSqlPool", {
 *     synapseWorkspaceId: exampleWorkspace.id,
 *     skuName: "DW100c",
 *     createMode: "Default",
 * });
 * const exampleSqlPoolWorkloadGroup = new azure.synapse.SqlPoolWorkloadGroup("exampleSqlPoolWorkloadGroup", {
 *     sqlPoolId: exampleSqlPool.id,
 *     importance: "normal",
 *     maxResourcePercent: 100,
 *     minResourcePercent: 0,
 *     maxResourcePercentPerRequest: 3,
 *     minResourcePercentPerRequest: 3,
 *     queryExecutionTimeoutInSeconds: 0,
 * });
 * const exampleSqlPoolWorkloadClassifier = new azure.synapse.SqlPoolWorkloadClassifier("exampleSqlPoolWorkloadClassifier", {
 *     workloadGroupId: exampleSqlPoolWorkloadGroup.id,
 *     context: "example_context",
 *     endTime: "14:00",
 *     importance: "high",
 *     label: "example_label",
 *     memberName: "dbo",
 *     startTime: "12:00",
 * });
 * ```
 *
 * ## Import
 *
 * Synapse SQL Pool Workload Classifiers can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:synapse/sqlPoolWorkloadClassifier:SqlPoolWorkloadClassifier example /subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/resGroup1/providers/Microsoft.Synapse/workspaces/workspace1/sqlPools/sqlPool1/workloadGroups/workloadGroup1/workloadClassifiers/workloadClassifier1
 * ```
 */
class SqlPoolWorkloadClassifier extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["context"] = state ? state.context : undefined;
            resourceInputs["endTime"] = state ? state.endTime : undefined;
            resourceInputs["importance"] = state ? state.importance : undefined;
            resourceInputs["label"] = state ? state.label : undefined;
            resourceInputs["memberName"] = state ? state.memberName : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["startTime"] = state ? state.startTime : undefined;
            resourceInputs["workloadGroupId"] = state ? state.workloadGroupId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.memberName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'memberName'");
            }
            if ((!args || args.workloadGroupId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workloadGroupId'");
            }
            resourceInputs["context"] = args ? args.context : undefined;
            resourceInputs["endTime"] = args ? args.endTime : undefined;
            resourceInputs["importance"] = args ? args.importance : undefined;
            resourceInputs["label"] = args ? args.label : undefined;
            resourceInputs["memberName"] = args ? args.memberName : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["startTime"] = args ? args.startTime : undefined;
            resourceInputs["workloadGroupId"] = args ? args.workloadGroupId : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(SqlPoolWorkloadClassifier.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing SqlPoolWorkloadClassifier resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new SqlPoolWorkloadClassifier(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SqlPoolWorkloadClassifier.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SqlPoolWorkloadClassifier.__pulumiType;
    }
}
exports.SqlPoolWorkloadClassifier = SqlPoolWorkloadClassifier;
/** @internal */
SqlPoolWorkloadClassifier.__pulumiType = 'azure:synapse/sqlPoolWorkloadClassifier:SqlPoolWorkloadClassifier';
