// @pulumi/azure/appconfiguration/configurationFeature.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConfigurationFeature = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an Azure App Configuration Feature.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const rg = new azure.core.ResourceGroup("rg", {location: "West Europe"});
 * const appconf = new azure.appconfiguration.ConfigurationStore("appconf", {
 *     resourceGroupName: rg.name,
 *     location: rg.location,
 * });
 * const test = new azure.appconfiguration.ConfigurationFeature("test", {
 *     configurationStoreId: appconf.id,
 *     description: "test description",
 *     label: `acctest-ackeylabel-%d`,
 *     enabled: true,
 * });
 * ```
 *
 * ## Import
 *
 * App Configuration Features can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:appconfiguration/configurationFeature:ConfigurationFeature test /subscriptions/00000000-0000-0000-0000-000000000000/resourcegroups/resourceGroup1/providers/Microsoft.AppConfiguration/configurationStores/appConf1/AppConfigurationFeature/appConfFeature1/Label/label1
 * ```
 *
 *  If you wish to import a key with an empty label then sustitute the label's name with `%00`, like this
 *
 * ```sh
 *  $ pulumi import azure:appconfiguration/configurationFeature:ConfigurationFeature test /subscriptions/00000000-0000-0000-0000-000000000000/resourcegroups/resourceGroup1/providers/Microsoft.AppConfiguration/configurationStores/appConf1/AppConfigurationFeature/appConfFeature1/Label/%00
 * ```
 */
class ConfigurationFeature extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["configurationStoreId"] = state ? state.configurationStoreId : undefined;
            resourceInputs["description"] = state ? state.description : undefined;
            resourceInputs["enabled"] = state ? state.enabled : undefined;
            resourceInputs["etag"] = state ? state.etag : undefined;
            resourceInputs["label"] = state ? state.label : undefined;
            resourceInputs["locked"] = state ? state.locked : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["percentageFilterValue"] = state ? state.percentageFilterValue : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
            resourceInputs["targetingFilters"] = state ? state.targetingFilters : undefined;
            resourceInputs["timewindowFilters"] = state ? state.timewindowFilters : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.configurationStoreId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'configurationStoreId'");
            }
            resourceInputs["configurationStoreId"] = args ? args.configurationStoreId : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["enabled"] = args ? args.enabled : undefined;
            resourceInputs["etag"] = args ? args.etag : undefined;
            resourceInputs["label"] = args ? args.label : undefined;
            resourceInputs["locked"] = args ? args.locked : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["percentageFilterValue"] = args ? args.percentageFilterValue : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["targetingFilters"] = args ? args.targetingFilters : undefined;
            resourceInputs["timewindowFilters"] = args ? args.timewindowFilters : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ConfigurationFeature.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing ConfigurationFeature resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ConfigurationFeature(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ConfigurationFeature.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ConfigurationFeature.__pulumiType;
    }
}
exports.ConfigurationFeature = ConfigurationFeature;
/** @internal */
ConfigurationFeature.__pulumiType = 'azure:appconfiguration/configurationFeature:ConfigurationFeature';
