// @pulumi/azure/storage/dataLakeGen2Path.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataLakeGen2Path = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Data Lake Gen2 Path in a File System within an Azure Storage Account.
 *
 * > **NOTE:** This resource requires some `Storage` specific roles which are not granted by default. Some of the built-ins roles that can be attributed are [`Storage Account Contributor`](https://docs.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#storage-account-contributor), [`Storage Blob Data Owner`](https://docs.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#storage-blob-data-owner), [`Storage Blob Data Contributor`](https://docs.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#storage-blob-data-contributor), [`Storage Blob Data Reader`](https://docs.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#storage-blob-data-reader).
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const exampleResourceGroup = new azure.core.ResourceGroup("exampleResourceGroup", {location: "West Europe"});
 * const exampleAccount = new azure.storage.Account("exampleAccount", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     location: exampleResourceGroup.location,
 *     accountTier: "Standard",
 *     accountReplicationType: "LRS",
 *     accountKind: "StorageV2",
 *     isHnsEnabled: "true",
 * });
 * const exampleDataLakeGen2Filesystem = new azure.storage.DataLakeGen2Filesystem("exampleDataLakeGen2Filesystem", {storageAccountId: exampleAccount.id});
 * const exampleDataLakeGen2Path = new azure.storage.DataLakeGen2Path("exampleDataLakeGen2Path", {
 *     path: "example",
 *     filesystemName: exampleDataLakeGen2Filesystem.name,
 *     storageAccountId: exampleAccount.id,
 *     resource: "directory",
 * });
 * ```
 *
 * ## Import
 *
 * Data Lake Gen2 Paths can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:storage/dataLakeGen2Path:DataLakeGen2Path example https://account1.dfs.core.windows.net/fileSystem1/path
 * ```
 */
class DataLakeGen2Path extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["aces"] = state ? state.aces : undefined;
            resourceInputs["filesystemName"] = state ? state.filesystemName : undefined;
            resourceInputs["group"] = state ? state.group : undefined;
            resourceInputs["owner"] = state ? state.owner : undefined;
            resourceInputs["path"] = state ? state.path : undefined;
            resourceInputs["resource"] = state ? state.resource : undefined;
            resourceInputs["storageAccountId"] = state ? state.storageAccountId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.filesystemName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'filesystemName'");
            }
            if ((!args || args.path === undefined) && !opts.urn) {
                throw new Error("Missing required property 'path'");
            }
            if ((!args || args.resource === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resource'");
            }
            if ((!args || args.storageAccountId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'storageAccountId'");
            }
            resourceInputs["aces"] = args ? args.aces : undefined;
            resourceInputs["filesystemName"] = args ? args.filesystemName : undefined;
            resourceInputs["group"] = args ? args.group : undefined;
            resourceInputs["owner"] = args ? args.owner : undefined;
            resourceInputs["path"] = args ? args.path : undefined;
            resourceInputs["resource"] = args ? args.resource : undefined;
            resourceInputs["storageAccountId"] = args ? args.storageAccountId : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(DataLakeGen2Path.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing DataLakeGen2Path resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new DataLakeGen2Path(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DataLakeGen2Path.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DataLakeGen2Path.__pulumiType;
    }
}
exports.DataLakeGen2Path = DataLakeGen2Path;
/** @internal */
DataLakeGen2Path.__pulumiType = 'azure:storage/dataLakeGen2Path:DataLakeGen2Path';
