// @pulumi/azure/iot/iotHubCertificate.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.IotHubCertificate = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an IotHub Device Provisioning Service Certificate.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 * import * from "fs";
 *
 * const exampleResourceGroup = new azure.core.ResourceGroup("exampleResourceGroup", {location: "West Europe"});
 * const exampleIotHubDps = new azure.iot.IotHubDps("exampleIotHubDps", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     location: exampleResourceGroup.location,
 *     sku: {
 *         name: "S1",
 *         capacity: "1",
 *     },
 * });
 * const exampleIotHubCertificate = new azure.iot.IotHubCertificate("exampleIotHubCertificate", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     iotDpsName: exampleIotHubDps.name,
 *     certificateContent: Buffer.from(fs.readFileSync("example.cer"), 'binary').toString('base64'),
 * });
 * ```
 *
 * ## Import
 *
 * IoTHub Device Provisioning Service Certificates can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:iot/iotHubCertificate:IotHubCertificate example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/Microsoft.Devices/provisioningServices/example/certificates/example
 * ```
 */
class IotHubCertificate extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["certificateContent"] = state ? state.certificateContent : undefined;
            resourceInputs["iotDpsName"] = state ? state.iotDpsName : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["resourceGroupName"] = state ? state.resourceGroupName : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.certificateContent === undefined) && !opts.urn) {
                throw new Error("Missing required property 'certificateContent'");
            }
            if ((!args || args.iotDpsName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'iotDpsName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["certificateContent"] = args ? args.certificateContent : undefined;
            resourceInputs["iotDpsName"] = args ? args.iotDpsName : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(IotHubCertificate.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing IotHubCertificate resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new IotHubCertificate(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of IotHubCertificate.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === IotHubCertificate.__pulumiType;
    }
}
exports.IotHubCertificate = IotHubCertificate;
/** @internal */
IotHubCertificate.__pulumiType = 'azure:iot/iotHubCertificate:IotHubCertificate';
