// @pulumi/azure/iot/securitySolution.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecuritySolution = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an iot security solution.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const exampleResourceGroup = new azure.core.ResourceGroup("exampleResourceGroup", {location: "West Europe"});
 * const exampleIoTHub = new azure.iot.IoTHub("exampleIoTHub", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     location: exampleResourceGroup.location,
 *     sku: {
 *         name: "S1",
 *         capacity: "1",
 *     },
 * });
 * const exampleSecuritySolution = new azure.iot.SecuritySolution("exampleSecuritySolution", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     location: exampleResourceGroup.location,
 *     displayName: "Iot Security Solution",
 *     iothubIds: [exampleIoTHub.id],
 * });
 * ```
 *
 * ## Import
 *
 * Iot Security Solution can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:iot/securitySolution:SecuritySolution example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resGroup1/providers/Microsoft.Security/IoTSecuritySolutions/solution1
 * ```
 */
class SecuritySolution extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["additionalWorkspaces"] = state ? state.additionalWorkspaces : undefined;
            resourceInputs["disabledDataSources"] = state ? state.disabledDataSources : undefined;
            resourceInputs["displayName"] = state ? state.displayName : undefined;
            resourceInputs["enabled"] = state ? state.enabled : undefined;
            resourceInputs["eventsToExports"] = state ? state.eventsToExports : undefined;
            resourceInputs["iothubIds"] = state ? state.iothubIds : undefined;
            resourceInputs["location"] = state ? state.location : undefined;
            resourceInputs["logAnalyticsWorkspaceId"] = state ? state.logAnalyticsWorkspaceId : undefined;
            resourceInputs["logUnmaskedIpsEnabled"] = state ? state.logUnmaskedIpsEnabled : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["queryForResources"] = state ? state.queryForResources : undefined;
            resourceInputs["querySubscriptionIds"] = state ? state.querySubscriptionIds : undefined;
            resourceInputs["recommendationsEnabled"] = state ? state.recommendationsEnabled : undefined;
            resourceInputs["resourceGroupName"] = state ? state.resourceGroupName : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.displayName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'displayName'");
            }
            if ((!args || args.iothubIds === undefined) && !opts.urn) {
                throw new Error("Missing required property 'iothubIds'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["additionalWorkspaces"] = args ? args.additionalWorkspaces : undefined;
            resourceInputs["disabledDataSources"] = args ? args.disabledDataSources : undefined;
            resourceInputs["displayName"] = args ? args.displayName : undefined;
            resourceInputs["enabled"] = args ? args.enabled : undefined;
            resourceInputs["eventsToExports"] = args ? args.eventsToExports : undefined;
            resourceInputs["iothubIds"] = args ? args.iothubIds : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["logAnalyticsWorkspaceId"] = args ? args.logAnalyticsWorkspaceId : undefined;
            resourceInputs["logUnmaskedIpsEnabled"] = args ? args.logUnmaskedIpsEnabled : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["queryForResources"] = args ? args.queryForResources : undefined;
            resourceInputs["querySubscriptionIds"] = args ? args.querySubscriptionIds : undefined;
            resourceInputs["recommendationsEnabled"] = args ? args.recommendationsEnabled : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(SecuritySolution.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing SecuritySolution resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new SecuritySolution(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SecuritySolution.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SecuritySolution.__pulumiType;
    }
}
exports.SecuritySolution = SecuritySolution;
/** @internal */
SecuritySolution.__pulumiType = 'azure:iot/securitySolution:SecuritySolution';
