// @pulumi/azure/config/vars.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
const __config = new pulumi.Config("azure");
Object.defineProperty(exports, "auxiliaryTenantIds", {
    get() {
        return __config.getObject("auxiliaryTenantIds");
    },
    enumerable: true,
});
Object.defineProperty(exports, "clientCertificatePassword", {
    get() {
        return __config.get("clientCertificatePassword");
    },
    enumerable: true,
});
Object.defineProperty(exports, "clientCertificatePath", {
    get() {
        return __config.get("clientCertificatePath");
    },
    enumerable: true,
});
Object.defineProperty(exports, "clientId", {
    get() {
        return __config.get("clientId");
    },
    enumerable: true,
});
Object.defineProperty(exports, "clientSecret", {
    get() {
        return __config.get("clientSecret");
    },
    enumerable: true,
});
Object.defineProperty(exports, "disableCorrelationRequestId", {
    get() {
        return __config.getObject("disableCorrelationRequestId");
    },
    enumerable: true,
});
Object.defineProperty(exports, "disableTerraformPartnerId", {
    get() {
        return __config.getObject("disableTerraformPartnerId");
    },
    enumerable: true,
});
Object.defineProperty(exports, "environment", {
    get() {
        var _a;
        return (_a = __config.get("environment")) !== null && _a !== void 0 ? _a : (utilities.getEnv("AZURE_ENVIRONMENT", "ARM_ENVIRONMENT") || "public");
    },
    enumerable: true,
});
Object.defineProperty(exports, "features", {
    get() {
        return __config.getObject("features");
    },
    enumerable: true,
});
Object.defineProperty(exports, "location", {
    get() {
        var _a;
        return (_a = __config.get("location")) !== null && _a !== void 0 ? _a : utilities.getEnv("ARM_LOCATION");
    },
    enumerable: true,
});
Object.defineProperty(exports, "metadataHost", {
    get() {
        var _a;
        return (_a = __config.get("metadataHost")) !== null && _a !== void 0 ? _a : utilities.getEnv("ARM_METADATA_HOSTNAME");
    },
    enumerable: true,
});
Object.defineProperty(exports, "metadataUrl", {
    get() {
        var _a;
        return (_a = __config.get("metadataUrl")) !== null && _a !== void 0 ? _a : utilities.getEnv("ARM_METADATA_URL");
    },
    enumerable: true,
});
Object.defineProperty(exports, "msiEndpoint", {
    get() {
        return __config.get("msiEndpoint");
    },
    enumerable: true,
});
Object.defineProperty(exports, "partnerId", {
    get() {
        return __config.get("partnerId");
    },
    enumerable: true,
});
Object.defineProperty(exports, "skipCredentialsValidation", {
    get() {
        return __config.getObject("skipCredentialsValidation");
    },
    enumerable: true,
});
Object.defineProperty(exports, "skipProviderRegistration", {
    get() {
        var _a;
        return (_a = __config.getObject("skipProviderRegistration")) !== null && _a !== void 0 ? _a : (utilities.getEnvBoolean("ARM_SKIP_PROVIDER_REGISTRATION") || false);
    },
    enumerable: true,
});
Object.defineProperty(exports, "storageUseAzuread", {
    get() {
        var _a;
        return (_a = __config.getObject("storageUseAzuread")) !== null && _a !== void 0 ? _a : (utilities.getEnvBoolean("ARM_STORAGE_USE_AZUREAD") || false);
    },
    enumerable: true,
});
Object.defineProperty(exports, "subscriptionId", {
    get() {
        var _a;
        return (_a = __config.get("subscriptionId")) !== null && _a !== void 0 ? _a : (utilities.getEnv("ARM_SUBSCRIPTION_ID") || "");
    },
    enumerable: true,
});
Object.defineProperty(exports, "tenantId", {
    get() {
        return __config.get("tenantId");
    },
    enumerable: true,
});
Object.defineProperty(exports, "useMsal", {
    get() {
        return __config.getObject("useMsal");
    },
    enumerable: true,
});
Object.defineProperty(exports, "useMsi", {
    get() {
        return __config.getObject("useMsi");
    },
    enumerable: true,
});
