// @pulumi/azure/compute/diskPoolIscsiTarget.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DiskPoolIscsiTarget = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an iSCSI Target.
 *
 * !> **Note:** Each Disk Pool can have a maximum of 1 iSCSI Target.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 * import * as azuread from "@pulumi/azuread";
 *
 * const exampleResourceGroup = new azure.core.ResourceGroup("exampleResourceGroup", {location: "West Europe"});
 * const exampleVirtualNetwork = new azure.network.VirtualNetwork("exampleVirtualNetwork", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     location: exampleResourceGroup.location,
 *     addressSpaces: ["10.0.0.0/16"],
 * });
 * const exampleSubnet = new azure.network.Subnet("exampleSubnet", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     virtualNetworkName: exampleVirtualNetwork.name,
 *     addressPrefixes: ["10.0.0.0/24"],
 *     delegations: [{
 *         name: "diskspool",
 *         serviceDelegation: {
 *             actions: ["Microsoft.Network/virtualNetworks/read"],
 *             name: "Microsoft.StoragePool/diskPools",
 *         },
 *     }],
 * });
 * const exampleDiskPool = new azure.compute.DiskPool("exampleDiskPool", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     location: exampleResourceGroup.location,
 *     subnetId: exampleSubnet.id,
 *     zones: ["1"],
 *     skuName: "Basic_B1",
 * });
 * const exampleManagedDisk = new azure.compute.ManagedDisk("exampleManagedDisk", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     location: exampleResourceGroup.location,
 *     createOption: "Empty",
 *     storageAccountType: "Premium_LRS",
 *     diskSizeGb: 4,
 *     maxShares: 2,
 *     zones: ["1"],
 * });
 * const exampleServicePrincipal = azuread.getServicePrincipal({
 *     displayName: "StoragePool Resource Provider",
 * });
 * const roles = [
 *     "Disk Pool Operator",
 *     "Virtual Machine Contributor",
 * ];
 * const exampleAssignment: azure.authorization.Assignment[];
 * for (const range = {value: 0}; range.value < roles.length; range.value++) {
 *     exampleAssignment.push(new azure.authorization.Assignment(`exampleAssignment-${range.value}`, {
 *         principalId: exampleServicePrincipal.then(exampleServicePrincipal => exampleServicePrincipal.id),
 *         roleDefinitionName: roles[range.value],
 *         scope: exampleManagedDisk.id,
 *     }));
 * }
 * const exampleDiskPoolManagedDiskAttachment = new azure.compute.DiskPoolManagedDiskAttachment("exampleDiskPoolManagedDiskAttachment", {
 *     diskPoolId: exampleDiskPool.id,
 *     managedDiskId: exampleManagedDisk.id,
 * }, {
 *     dependsOn: [exampleAssignment],
 * });
 * const exampleDiskPoolIscsiTarget = new azure.compute.DiskPoolIscsiTarget("exampleDiskPoolIscsiTarget", {
 *     aclMode: "Dynamic",
 *     disksPoolId: exampleDiskPool.id,
 *     targetIqn: "iqn.2021-11.com.microsoft:test",
 * }, {
 *     dependsOn: [exampleDiskPoolManagedDiskAttachment],
 * });
 * ```
 *
 * ## Import
 *
 * iSCSI Targets can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:compute/diskPoolIscsiTarget:DiskPoolIscsiTarget example /subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/group1/providers/Microsoft.StoragePool/diskPools/pool1/iscsiTargets/iscsiTarget1
 * ```
 */
class DiskPoolIscsiTarget extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["aclMode"] = state ? state.aclMode : undefined;
            resourceInputs["disksPoolId"] = state ? state.disksPoolId : undefined;
            resourceInputs["endpoints"] = state ? state.endpoints : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["port"] = state ? state.port : undefined;
            resourceInputs["targetIqn"] = state ? state.targetIqn : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.aclMode === undefined) && !opts.urn) {
                throw new Error("Missing required property 'aclMode'");
            }
            if ((!args || args.disksPoolId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'disksPoolId'");
            }
            resourceInputs["aclMode"] = args ? args.aclMode : undefined;
            resourceInputs["disksPoolId"] = args ? args.disksPoolId : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["targetIqn"] = args ? args.targetIqn : undefined;
            resourceInputs["endpoints"] = undefined /*out*/;
            resourceInputs["port"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(DiskPoolIscsiTarget.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing DiskPoolIscsiTarget resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new DiskPoolIscsiTarget(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DiskPoolIscsiTarget.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DiskPoolIscsiTarget.__pulumiType;
    }
}
exports.DiskPoolIscsiTarget = DiskPoolIscsiTarget;
/** @internal */
DiskPoolIscsiTarget.__pulumiType = 'azure:compute/diskPoolIscsiTarget:DiskPoolIscsiTarget';
