// @pulumi/azure/containerservice/kubernetesClusterNodePool.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.KubernetesClusterNodePool = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * ## Import
 *
 * Kubernetes Cluster Node Pools can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:containerservice/kubernetesClusterNodePool:KubernetesClusterNodePool pool1 /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.ContainerService/managedClusters/cluster1/agentPools/pool1
 * ```
 */
class KubernetesClusterNodePool extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            inputs["availabilityZones"] = state ? state.availabilityZones : undefined;
            inputs["enableAutoScaling"] = state ? state.enableAutoScaling : undefined;
            inputs["enableHostEncryption"] = state ? state.enableHostEncryption : undefined;
            inputs["enableNodePublicIp"] = state ? state.enableNodePublicIp : undefined;
            inputs["evictionPolicy"] = state ? state.evictionPolicy : undefined;
            inputs["fipsEnabled"] = state ? state.fipsEnabled : undefined;
            inputs["kubeletConfig"] = state ? state.kubeletConfig : undefined;
            inputs["kubeletDiskType"] = state ? state.kubeletDiskType : undefined;
            inputs["kubernetesClusterId"] = state ? state.kubernetesClusterId : undefined;
            inputs["linuxOsConfig"] = state ? state.linuxOsConfig : undefined;
            inputs["maxCount"] = state ? state.maxCount : undefined;
            inputs["maxPods"] = state ? state.maxPods : undefined;
            inputs["minCount"] = state ? state.minCount : undefined;
            inputs["mode"] = state ? state.mode : undefined;
            inputs["name"] = state ? state.name : undefined;
            inputs["nodeCount"] = state ? state.nodeCount : undefined;
            inputs["nodeLabels"] = state ? state.nodeLabels : undefined;
            inputs["nodePublicIpPrefixId"] = state ? state.nodePublicIpPrefixId : undefined;
            inputs["nodeTaints"] = state ? state.nodeTaints : undefined;
            inputs["orchestratorVersion"] = state ? state.orchestratorVersion : undefined;
            inputs["osDiskSizeGb"] = state ? state.osDiskSizeGb : undefined;
            inputs["osDiskType"] = state ? state.osDiskType : undefined;
            inputs["osSku"] = state ? state.osSku : undefined;
            inputs["osType"] = state ? state.osType : undefined;
            inputs["podSubnetId"] = state ? state.podSubnetId : undefined;
            inputs["priority"] = state ? state.priority : undefined;
            inputs["proximityPlacementGroupId"] = state ? state.proximityPlacementGroupId : undefined;
            inputs["scaleDownMode"] = state ? state.scaleDownMode : undefined;
            inputs["spotMaxPrice"] = state ? state.spotMaxPrice : undefined;
            inputs["tags"] = state ? state.tags : undefined;
            inputs["ultraSsdEnabled"] = state ? state.ultraSsdEnabled : undefined;
            inputs["upgradeSettings"] = state ? state.upgradeSettings : undefined;
            inputs["vmSize"] = state ? state.vmSize : undefined;
            inputs["vnetSubnetId"] = state ? state.vnetSubnetId : undefined;
            inputs["workloadRuntime"] = state ? state.workloadRuntime : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.kubernetesClusterId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kubernetesClusterId'");
            }
            if ((!args || args.vmSize === undefined) && !opts.urn) {
                throw new Error("Missing required property 'vmSize'");
            }
            inputs["availabilityZones"] = args ? args.availabilityZones : undefined;
            inputs["enableAutoScaling"] = args ? args.enableAutoScaling : undefined;
            inputs["enableHostEncryption"] = args ? args.enableHostEncryption : undefined;
            inputs["enableNodePublicIp"] = args ? args.enableNodePublicIp : undefined;
            inputs["evictionPolicy"] = args ? args.evictionPolicy : undefined;
            inputs["fipsEnabled"] = args ? args.fipsEnabled : undefined;
            inputs["kubeletConfig"] = args ? args.kubeletConfig : undefined;
            inputs["kubeletDiskType"] = args ? args.kubeletDiskType : undefined;
            inputs["kubernetesClusterId"] = args ? args.kubernetesClusterId : undefined;
            inputs["linuxOsConfig"] = args ? args.linuxOsConfig : undefined;
            inputs["maxCount"] = args ? args.maxCount : undefined;
            inputs["maxPods"] = args ? args.maxPods : undefined;
            inputs["minCount"] = args ? args.minCount : undefined;
            inputs["mode"] = args ? args.mode : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["nodeCount"] = args ? args.nodeCount : undefined;
            inputs["nodeLabels"] = args ? args.nodeLabels : undefined;
            inputs["nodePublicIpPrefixId"] = args ? args.nodePublicIpPrefixId : undefined;
            inputs["nodeTaints"] = args ? args.nodeTaints : undefined;
            inputs["orchestratorVersion"] = args ? args.orchestratorVersion : undefined;
            inputs["osDiskSizeGb"] = args ? args.osDiskSizeGb : undefined;
            inputs["osDiskType"] = args ? args.osDiskType : undefined;
            inputs["osSku"] = args ? args.osSku : undefined;
            inputs["osType"] = args ? args.osType : undefined;
            inputs["podSubnetId"] = args ? args.podSubnetId : undefined;
            inputs["priority"] = args ? args.priority : undefined;
            inputs["proximityPlacementGroupId"] = args ? args.proximityPlacementGroupId : undefined;
            inputs["scaleDownMode"] = args ? args.scaleDownMode : undefined;
            inputs["spotMaxPrice"] = args ? args.spotMaxPrice : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["ultraSsdEnabled"] = args ? args.ultraSsdEnabled : undefined;
            inputs["upgradeSettings"] = args ? args.upgradeSettings : undefined;
            inputs["vmSize"] = args ? args.vmSize : undefined;
            inputs["vnetSubnetId"] = args ? args.vnetSubnetId : undefined;
            inputs["workloadRuntime"] = args ? args.workloadRuntime : undefined;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(KubernetesClusterNodePool.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing KubernetesClusterNodePool resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new KubernetesClusterNodePool(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of KubernetesClusterNodePool.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === KubernetesClusterNodePool.__pulumiType;
    }
}
exports.KubernetesClusterNodePool = KubernetesClusterNodePool;
/** @internal */
KubernetesClusterNodePool.__pulumiType = 'azure:containerservice/kubernetesClusterNodePool:KubernetesClusterNodePool';
