// @pulumi/azure/containerservice/registry.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Registry = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an Azure Container Registry.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const rg = new azure.core.ResourceGroup("rg", {location: "West Europe"});
 * const acr = new azure.containerservice.Registry("acr", {
 *     resourceGroupName: rg.name,
 *     location: rg.location,
 *     sku: "Premium",
 *     adminEnabled: false,
 *     georeplications: [
 *         {
 *             location: "East US",
 *             zoneRedundancyEnabled: true,
 *             tags: {},
 *         },
 *         {
 *             location: "westeurope",
 *             zoneRedundancyEnabled: true,
 *             tags: {},
 *         },
 *     ],
 * });
 * ```
 * ### Encryption)
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const rg = new azure.core.ResourceGroup("rg", {location: "West Europe"});
 * const exampleUserAssignedIdentity = new azure.authorization.UserAssignedIdentity("exampleUserAssignedIdentity", {
 *     resourceGroupName: azurerm_resource_group.example.name,
 *     location: azurerm_resource_group.example.location,
 * });
 * const exampleKey = azure.keyvault.getKey({
 *     name: "super-secret",
 *     keyVaultId: data.azurerm_key_vault.existing.id,
 * });
 * const acr = new azure.containerservice.Registry("acr", {
 *     resourceGroupName: rg.name,
 *     location: rg.location,
 *     sku: "Premium",
 *     identity: {
 *         type: "UserAssigned",
 *         identityIds: [exampleUserAssignedIdentity.id],
 *     },
 *     encryption: {
 *         enabled: true,
 *         keyVaultKeyId: exampleKey.then(exampleKey => exampleKey.id),
 *         identityClientId: exampleUserAssignedIdentity.clientId,
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Container Registries can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:containerservice/registry:Registry example /subscriptions/00000000-0000-0000-0000-000000000000/resourcegroups/mygroup1/providers/Microsoft.ContainerRegistry/registries/myregistry1
 * ```
 */
class Registry extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            inputs["adminEnabled"] = state ? state.adminEnabled : undefined;
            inputs["adminPassword"] = state ? state.adminPassword : undefined;
            inputs["adminUsername"] = state ? state.adminUsername : undefined;
            inputs["anonymousPullEnabled"] = state ? state.anonymousPullEnabled : undefined;
            inputs["dataEndpointEnabled"] = state ? state.dataEndpointEnabled : undefined;
            inputs["encryption"] = state ? state.encryption : undefined;
            inputs["georeplicationLocations"] = state ? state.georeplicationLocations : undefined;
            inputs["georeplications"] = state ? state.georeplications : undefined;
            inputs["identity"] = state ? state.identity : undefined;
            inputs["location"] = state ? state.location : undefined;
            inputs["loginServer"] = state ? state.loginServer : undefined;
            inputs["name"] = state ? state.name : undefined;
            inputs["networkRuleBypassOption"] = state ? state.networkRuleBypassOption : undefined;
            inputs["networkRuleSet"] = state ? state.networkRuleSet : undefined;
            inputs["publicNetworkAccessEnabled"] = state ? state.publicNetworkAccessEnabled : undefined;
            inputs["quarantinePolicyEnabled"] = state ? state.quarantinePolicyEnabled : undefined;
            inputs["resourceGroupName"] = state ? state.resourceGroupName : undefined;
            inputs["retentionPolicy"] = state ? state.retentionPolicy : undefined;
            inputs["sku"] = state ? state.sku : undefined;
            inputs["storageAccountId"] = state ? state.storageAccountId : undefined;
            inputs["tags"] = state ? state.tags : undefined;
            inputs["trustPolicy"] = state ? state.trustPolicy : undefined;
            inputs["zoneRedundancyEnabled"] = state ? state.zoneRedundancyEnabled : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["adminEnabled"] = args ? args.adminEnabled : undefined;
            inputs["anonymousPullEnabled"] = args ? args.anonymousPullEnabled : undefined;
            inputs["dataEndpointEnabled"] = args ? args.dataEndpointEnabled : undefined;
            inputs["encryption"] = args ? args.encryption : undefined;
            inputs["georeplicationLocations"] = args ? args.georeplicationLocations : undefined;
            inputs["georeplications"] = args ? args.georeplications : undefined;
            inputs["identity"] = args ? args.identity : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["networkRuleBypassOption"] = args ? args.networkRuleBypassOption : undefined;
            inputs["networkRuleSet"] = args ? args.networkRuleSet : undefined;
            inputs["publicNetworkAccessEnabled"] = args ? args.publicNetworkAccessEnabled : undefined;
            inputs["quarantinePolicyEnabled"] = args ? args.quarantinePolicyEnabled : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["retentionPolicy"] = args ? args.retentionPolicy : undefined;
            inputs["sku"] = args ? args.sku : undefined;
            inputs["storageAccountId"] = args ? args.storageAccountId : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["trustPolicy"] = args ? args.trustPolicy : undefined;
            inputs["zoneRedundancyEnabled"] = args ? args.zoneRedundancyEnabled : undefined;
            inputs["adminPassword"] = undefined /*out*/;
            inputs["adminUsername"] = undefined /*out*/;
            inputs["loginServer"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(Registry.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Registry resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Registry(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Registry.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Registry.__pulumiType;
    }
}
exports.Registry = Registry;
/** @internal */
Registry.__pulumiType = 'azure:containerservice/registry:Registry';
