// @pulumi/azure/datafactory/triggerBlobEvent.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TriggerBlobEvent = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Blob Event Trigger inside an Azure Data Factory.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const exampleResourceGroup = new azure.core.ResourceGroup("exampleResourceGroup", {location: "West Europe"});
 * const exampleFactory = new azure.datafactory.Factory("exampleFactory", {
 *     location: exampleResourceGroup.location,
 *     resourceGroupName: exampleResourceGroup.name,
 * });
 * const examplePipeline = new azure.datafactory.Pipeline("examplePipeline", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     dataFactoryId: exampleFactory.id,
 * });
 * const exampleAccount = new azure.storage.Account("exampleAccount", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     location: exampleResourceGroup.location,
 *     accountTier: "Standard",
 *     accountReplicationType: "LRS",
 * });
 * const exampleTriggerBlobEvent = new azure.datafactory.TriggerBlobEvent("exampleTriggerBlobEvent", {
 *     dataFactoryId: exampleFactory.id,
 *     storageAccountId: exampleAccount.id,
 *     events: [
 *         "Microsoft.Storage.BlobCreated",
 *         "Microsoft.Storage.BlobDeleted",
 *     ],
 *     blobPathEndsWith: ".txt",
 *     ignoreEmptyBlobs: true,
 *     activated: true,
 *     annotations: [
 *         "test1",
 *         "test2",
 *         "test3",
 *     ],
 *     description: "example description",
 *     pipelines: [{
 *         name: examplePipeline.name,
 *         parameters: {
 *             Env: "Prod",
 *         },
 *     }],
 *     additionalProperties: {
 *         foo: "foo1",
 *         bar: "bar2",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Data Factory Blob Event Trigger can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:datafactory/triggerBlobEvent:TriggerBlobEvent example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/example/providers/Microsoft.DataFactory/factories/example/triggers/example
 * ```
 */
class TriggerBlobEvent extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            inputs["activated"] = state ? state.activated : undefined;
            inputs["additionalProperties"] = state ? state.additionalProperties : undefined;
            inputs["annotations"] = state ? state.annotations : undefined;
            inputs["blobPathBeginsWith"] = state ? state.blobPathBeginsWith : undefined;
            inputs["blobPathEndsWith"] = state ? state.blobPathEndsWith : undefined;
            inputs["dataFactoryId"] = state ? state.dataFactoryId : undefined;
            inputs["description"] = state ? state.description : undefined;
            inputs["events"] = state ? state.events : undefined;
            inputs["ignoreEmptyBlobs"] = state ? state.ignoreEmptyBlobs : undefined;
            inputs["name"] = state ? state.name : undefined;
            inputs["pipelines"] = state ? state.pipelines : undefined;
            inputs["storageAccountId"] = state ? state.storageAccountId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.dataFactoryId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dataFactoryId'");
            }
            if ((!args || args.events === undefined) && !opts.urn) {
                throw new Error("Missing required property 'events'");
            }
            if ((!args || args.pipelines === undefined) && !opts.urn) {
                throw new Error("Missing required property 'pipelines'");
            }
            if ((!args || args.storageAccountId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'storageAccountId'");
            }
            inputs["activated"] = args ? args.activated : undefined;
            inputs["additionalProperties"] = args ? args.additionalProperties : undefined;
            inputs["annotations"] = args ? args.annotations : undefined;
            inputs["blobPathBeginsWith"] = args ? args.blobPathBeginsWith : undefined;
            inputs["blobPathEndsWith"] = args ? args.blobPathEndsWith : undefined;
            inputs["dataFactoryId"] = args ? args.dataFactoryId : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["events"] = args ? args.events : undefined;
            inputs["ignoreEmptyBlobs"] = args ? args.ignoreEmptyBlobs : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["pipelines"] = args ? args.pipelines : undefined;
            inputs["storageAccountId"] = args ? args.storageAccountId : undefined;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(TriggerBlobEvent.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing TriggerBlobEvent resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new TriggerBlobEvent(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of TriggerBlobEvent.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TriggerBlobEvent.__pulumiType;
    }
}
exports.TriggerBlobEvent = TriggerBlobEvent;
/** @internal */
TriggerBlobEvent.__pulumiType = 'azure:datafactory/triggerBlobEvent:TriggerBlobEvent';
