// @pulumi/azure/network/expressRouteCircuitPeering.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExpressRouteCircuitPeering = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an ExpressRoute Circuit Peering.
 *
 * ## Example Usage
 * ### Creating A Microsoft Peering)
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const exampleResourceGroup = new azure.core.ResourceGroup("exampleResourceGroup", {location: "West Europe"});
 * const exampleExpressRouteCircuit = new azure.network.ExpressRouteCircuit("exampleExpressRouteCircuit", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     location: exampleResourceGroup.location,
 *     serviceProviderName: "Equinix",
 *     peeringLocation: "Silicon Valley",
 *     bandwidthInMbps: 50,
 *     sku: {
 *         tier: "Standard",
 *         family: "MeteredData",
 *     },
 *     allowClassicOperations: false,
 *     tags: {
 *         environment: "Production",
 *     },
 * });
 * const exampleExpressRouteCircuitPeering = new azure.network.ExpressRouteCircuitPeering("exampleExpressRouteCircuitPeering", {
 *     peeringType: "MicrosoftPeering",
 *     expressRouteCircuitName: exampleExpressRouteCircuit.name,
 *     resourceGroupName: exampleResourceGroup.name,
 *     peerAsn: 100,
 *     primaryPeerAddressPrefix: "123.0.0.0/30",
 *     secondaryPeerAddressPrefix: "123.0.0.4/30",
 *     vlanId: 300,
 *     microsoftPeeringConfig: {
 *         advertisedPublicPrefixes: ["123.1.0.0/24"],
 *     },
 *     ipv6: {
 *         primaryPeerAddressPrefix: "2002:db01::/126",
 *         secondaryPeerAddressPrefix: "2003:db01::/126",
 *         microsoftPeering: {
 *             advertisedPublicPrefixes: ["2002:db01::/126"],
 *         },
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * ExpressRoute Circuit Peerings can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:network/expressRouteCircuitPeering:ExpressRouteCircuitPeering peering1 /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/Microsoft.Network/expressRouteCircuits/myExpressRoute/peerings/peering1
 * ```
 */
class ExpressRouteCircuitPeering extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            inputs["azureAsn"] = state ? state.azureAsn : undefined;
            inputs["expressRouteCircuitName"] = state ? state.expressRouteCircuitName : undefined;
            inputs["ipv6"] = state ? state.ipv6 : undefined;
            inputs["microsoftPeeringConfig"] = state ? state.microsoftPeeringConfig : undefined;
            inputs["peerAsn"] = state ? state.peerAsn : undefined;
            inputs["peeringType"] = state ? state.peeringType : undefined;
            inputs["primaryAzurePort"] = state ? state.primaryAzurePort : undefined;
            inputs["primaryPeerAddressPrefix"] = state ? state.primaryPeerAddressPrefix : undefined;
            inputs["resourceGroupName"] = state ? state.resourceGroupName : undefined;
            inputs["routeFilterId"] = state ? state.routeFilterId : undefined;
            inputs["secondaryAzurePort"] = state ? state.secondaryAzurePort : undefined;
            inputs["secondaryPeerAddressPrefix"] = state ? state.secondaryPeerAddressPrefix : undefined;
            inputs["sharedKey"] = state ? state.sharedKey : undefined;
            inputs["vlanId"] = state ? state.vlanId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.expressRouteCircuitName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'expressRouteCircuitName'");
            }
            if ((!args || args.peeringType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'peeringType'");
            }
            if ((!args || args.primaryPeerAddressPrefix === undefined) && !opts.urn) {
                throw new Error("Missing required property 'primaryPeerAddressPrefix'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.secondaryPeerAddressPrefix === undefined) && !opts.urn) {
                throw new Error("Missing required property 'secondaryPeerAddressPrefix'");
            }
            if ((!args || args.vlanId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'vlanId'");
            }
            inputs["expressRouteCircuitName"] = args ? args.expressRouteCircuitName : undefined;
            inputs["ipv6"] = args ? args.ipv6 : undefined;
            inputs["microsoftPeeringConfig"] = args ? args.microsoftPeeringConfig : undefined;
            inputs["peerAsn"] = args ? args.peerAsn : undefined;
            inputs["peeringType"] = args ? args.peeringType : undefined;
            inputs["primaryPeerAddressPrefix"] = args ? args.primaryPeerAddressPrefix : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["routeFilterId"] = args ? args.routeFilterId : undefined;
            inputs["secondaryPeerAddressPrefix"] = args ? args.secondaryPeerAddressPrefix : undefined;
            inputs["sharedKey"] = args ? args.sharedKey : undefined;
            inputs["vlanId"] = args ? args.vlanId : undefined;
            inputs["azureAsn"] = undefined /*out*/;
            inputs["primaryAzurePort"] = undefined /*out*/;
            inputs["secondaryAzurePort"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(ExpressRouteCircuitPeering.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ExpressRouteCircuitPeering resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ExpressRouteCircuitPeering(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ExpressRouteCircuitPeering.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ExpressRouteCircuitPeering.__pulumiType;
    }
}
exports.ExpressRouteCircuitPeering = ExpressRouteCircuitPeering;
/** @internal */
ExpressRouteCircuitPeering.__pulumiType = 'azure:network/expressRouteCircuitPeering:ExpressRouteCircuitPeering';
