// @pulumi/azure/kusto/attachedDatabaseConfiguration.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AttachedDatabaseConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Kusto (also known as Azure Data Explorer) Attached Database Configuration
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const rg = new azure.core.ResourceGroup("rg", {location: "West Europe"});
 * const followerCluster = new azure.kusto.Cluster("followerCluster", {
 *     location: rg.location,
 *     resourceGroupName: rg.name,
 *     sku: {
 *         name: "Dev(No SLA)_Standard_D11_v2",
 *         capacity: 1,
 *     },
 * });
 * const followedCluster = new azure.kusto.Cluster("followedCluster", {
 *     location: rg.location,
 *     resourceGroupName: rg.name,
 *     sku: {
 *         name: "Dev(No SLA)_Standard_D11_v2",
 *         capacity: 1,
 *     },
 * });
 * const followedDatabase = new azure.kusto.Database("followedDatabase", {
 *     resourceGroupName: rg.name,
 *     location: rg.location,
 *     clusterName: azurerm_kusto_cluster.cluster2.name,
 * });
 * const exampleDatabase = new azure.kusto.Database("exampleDatabase", {
 *     resourceGroupName: rg.name,
 *     location: rg.location,
 *     clusterName: azurerm_kusto_cluster.cluster2.name,
 * });
 * const exampleAttachedDatabaseConfiguration = new azure.kusto.AttachedDatabaseConfiguration("exampleAttachedDatabaseConfiguration", {
 *     resourceGroupName: rg.name,
 *     location: rg.location,
 *     clusterName: followerCluster.name,
 *     clusterResourceId: followedCluster.id,
 *     databaseName: exampleDatabase.name,
 *     defaultPrincipalModificationsKind: "Union",
 *     sharing: {
 *         externalTablesToExcludes: ["ExternalTable2"],
 *         externalTablesToIncludes: ["ExternalTable1"],
 *         materializedViewsToExcludes: ["MaterializedViewTable2"],
 *         materializedViewsToIncludes: ["MaterializedViewTable1"],
 *         tablesToExcludes: ["Table2"],
 *         tablesToIncludes: ["Table1"],
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Kusto Attached Database Configurations can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:kusto/attachedDatabaseConfiguration:AttachedDatabaseConfiguration example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Kusto/Clusters/cluster1/AttachedDatabaseConfigurations/configuration1
 * ```
 */
class AttachedDatabaseConfiguration extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            inputs["attachedDatabaseNames"] = state ? state.attachedDatabaseNames : undefined;
            inputs["clusterName"] = state ? state.clusterName : undefined;
            inputs["clusterResourceId"] = state ? state.clusterResourceId : undefined;
            inputs["databaseName"] = state ? state.databaseName : undefined;
            inputs["defaultPrincipalModificationKind"] = state ? state.defaultPrincipalModificationKind : undefined;
            inputs["location"] = state ? state.location : undefined;
            inputs["name"] = state ? state.name : undefined;
            inputs["resourceGroupName"] = state ? state.resourceGroupName : undefined;
            inputs["sharing"] = state ? state.sharing : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.clusterName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'clusterName'");
            }
            if ((!args || args.clusterResourceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'clusterResourceId'");
            }
            if ((!args || args.databaseName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'databaseName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["clusterName"] = args ? args.clusterName : undefined;
            inputs["clusterResourceId"] = args ? args.clusterResourceId : undefined;
            inputs["databaseName"] = args ? args.databaseName : undefined;
            inputs["defaultPrincipalModificationKind"] = args ? args.defaultPrincipalModificationKind : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["sharing"] = args ? args.sharing : undefined;
            inputs["attachedDatabaseNames"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(AttachedDatabaseConfiguration.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing AttachedDatabaseConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AttachedDatabaseConfiguration(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AttachedDatabaseConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AttachedDatabaseConfiguration.__pulumiType;
    }
}
exports.AttachedDatabaseConfiguration = AttachedDatabaseConfiguration;
/** @internal */
AttachedDatabaseConfiguration.__pulumiType = 'azure:kusto/attachedDatabaseConfiguration:AttachedDatabaseConfiguration';
