// @pulumi/azure/eventgrid/eventSubscription.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventSubscription = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an EventGrid Event Subscription
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const defaultResourceGroup = new azure.core.ResourceGroup("defaultResourceGroup", {location: "West Europe"});
 * const defaultAccount = new azure.storage.Account("defaultAccount", {
 *     resourceGroupName: defaultResourceGroup.name,
 *     location: defaultResourceGroup.location,
 *     accountTier: "Standard",
 *     accountReplicationType: "LRS",
 *     tags: {
 *         environment: "staging",
 *     },
 * });
 * const defaultQueue = new azure.storage.Queue("defaultQueue", {storageAccountName: defaultAccount.name});
 * const defaultEventSubscription = new azure.eventgrid.EventSubscription("defaultEventSubscription", {
 *     scope: defaultResourceGroup.id,
 *     storageQueueEndpoint: {
 *         storageAccountId: defaultAccount.id,
 *         queueName: defaultQueue.name,
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * EventGrid Event Subscription's can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:eventgrid/eventSubscription:EventSubscription eventSubscription1
 * ```
 *
 *  /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.EventGrid/topics/topic1/providers/Microsoft.EventGrid/eventSubscriptions/eventSubscription1
 */
class EventSubscription extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["advancedFilter"] = state ? state.advancedFilter : undefined;
            resourceInputs["advancedFilteringOnArraysEnabled"] = state ? state.advancedFilteringOnArraysEnabled : undefined;
            resourceInputs["azureFunctionEndpoint"] = state ? state.azureFunctionEndpoint : undefined;
            resourceInputs["deadLetterIdentity"] = state ? state.deadLetterIdentity : undefined;
            resourceInputs["deliveryIdentity"] = state ? state.deliveryIdentity : undefined;
            resourceInputs["deliveryProperties"] = state ? state.deliveryProperties : undefined;
            resourceInputs["eventDeliverySchema"] = state ? state.eventDeliverySchema : undefined;
            resourceInputs["eventhubEndpoint"] = state ? state.eventhubEndpoint : undefined;
            resourceInputs["eventhubEndpointId"] = state ? state.eventhubEndpointId : undefined;
            resourceInputs["expirationTimeUtc"] = state ? state.expirationTimeUtc : undefined;
            resourceInputs["hybridConnectionEndpoint"] = state ? state.hybridConnectionEndpoint : undefined;
            resourceInputs["hybridConnectionEndpointId"] = state ? state.hybridConnectionEndpointId : undefined;
            resourceInputs["includedEventTypes"] = state ? state.includedEventTypes : undefined;
            resourceInputs["labels"] = state ? state.labels : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["retryPolicy"] = state ? state.retryPolicy : undefined;
            resourceInputs["scope"] = state ? state.scope : undefined;
            resourceInputs["serviceBusQueueEndpointId"] = state ? state.serviceBusQueueEndpointId : undefined;
            resourceInputs["serviceBusTopicEndpointId"] = state ? state.serviceBusTopicEndpointId : undefined;
            resourceInputs["storageBlobDeadLetterDestination"] = state ? state.storageBlobDeadLetterDestination : undefined;
            resourceInputs["storageQueueEndpoint"] = state ? state.storageQueueEndpoint : undefined;
            resourceInputs["subjectFilter"] = state ? state.subjectFilter : undefined;
            resourceInputs["topicName"] = state ? state.topicName : undefined;
            resourceInputs["webhookEndpoint"] = state ? state.webhookEndpoint : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.scope === undefined) && !opts.urn) {
                throw new Error("Missing required property 'scope'");
            }
            resourceInputs["advancedFilter"] = args ? args.advancedFilter : undefined;
            resourceInputs["advancedFilteringOnArraysEnabled"] = args ? args.advancedFilteringOnArraysEnabled : undefined;
            resourceInputs["azureFunctionEndpoint"] = args ? args.azureFunctionEndpoint : undefined;
            resourceInputs["deadLetterIdentity"] = args ? args.deadLetterIdentity : undefined;
            resourceInputs["deliveryIdentity"] = args ? args.deliveryIdentity : undefined;
            resourceInputs["deliveryProperties"] = args ? args.deliveryProperties : undefined;
            resourceInputs["eventDeliverySchema"] = args ? args.eventDeliverySchema : undefined;
            resourceInputs["eventhubEndpoint"] = args ? args.eventhubEndpoint : undefined;
            resourceInputs["eventhubEndpointId"] = args ? args.eventhubEndpointId : undefined;
            resourceInputs["expirationTimeUtc"] = args ? args.expirationTimeUtc : undefined;
            resourceInputs["hybridConnectionEndpoint"] = args ? args.hybridConnectionEndpoint : undefined;
            resourceInputs["hybridConnectionEndpointId"] = args ? args.hybridConnectionEndpointId : undefined;
            resourceInputs["includedEventTypes"] = args ? args.includedEventTypes : undefined;
            resourceInputs["labels"] = args ? args.labels : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["retryPolicy"] = args ? args.retryPolicy : undefined;
            resourceInputs["scope"] = args ? args.scope : undefined;
            resourceInputs["serviceBusQueueEndpointId"] = args ? args.serviceBusQueueEndpointId : undefined;
            resourceInputs["serviceBusTopicEndpointId"] = args ? args.serviceBusTopicEndpointId : undefined;
            resourceInputs["storageBlobDeadLetterDestination"] = args ? args.storageBlobDeadLetterDestination : undefined;
            resourceInputs["storageQueueEndpoint"] = args ? args.storageQueueEndpoint : undefined;
            resourceInputs["subjectFilter"] = args ? args.subjectFilter : undefined;
            resourceInputs["topicName"] = args ? args.topicName : undefined;
            resourceInputs["webhookEndpoint"] = args ? args.webhookEndpoint : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure:eventhub/eventSubscription:EventSubscription" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(EventSubscription.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing EventSubscription resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new EventSubscription(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of EventSubscription.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EventSubscription.__pulumiType;
    }
}
exports.EventSubscription = EventSubscription;
/** @internal */
EventSubscription.__pulumiType = 'azure:eventgrid/eventSubscription:EventSubscription';
